diff --git a/server/handlers.go b/server/handlers.go
index f8d0ed64..fd30bc34 100644
--- a/server/handlers.go
+++ b/server/handlers.go
@@ -581,7 +581,7 @@ func (s *Server) finalizeLogin(ctx context.Context, identity connector.Identity,
 	}
 
 	// we can skip the redirect to /approval and go ahead and send code if it's not required
-	if s.skipApproval && !authReq.ForceApprovalPrompt {
+	if s.skipApproval {
 		return "", true, nil
 	}
 
