diff --git a/server/handlers.go b/server/handlers.go
index 6521bf6a..74b0a819 100644
--- a/server/handlers.go
+++ b/server/handlers.go
@@ -535,7 +535,7 @@ func (s *Server) finalizeLogin(ctx context.Context, identity connector.Identity,
 		"preferred_username", claims.PreferredUsername, "email", email, "groups", claims.Groups)
 
 	// we can skip the redirect to /approval and go ahead and send code if it's not required
-	if s.skipApproval && !authReq.ForceApprovalPrompt {
+	if s.skipApproval {
 		return "", true, nil
 	}
 
