diff --git a/main.go b/main.go
index ab5c514..981dd5c 100644
--- a/main.go
+++ b/main.go
@@ -6,6 +6,7 @@ package main
 
 import (
 	"context"
+	"net"
 	"net/http"
 	"time"
 
@@ -38,6 +39,7 @@ var envs = []string{
 
 type config struct {
 	Address            string        `envconfig:"DRONE_BIND"`
+	UnixSocket         string        `envconfig:"DRONE_UNIX_SOCKET"`
 	Debug              bool          `envconfig:"DRONE_DEBUG"`
 	Secret             string        `envconfig:"DRONE_SECRET"`
 	DisallowForks      bool          `envconfig:"DRONE_DISALLOW_FORKS"`
@@ -137,10 +139,21 @@ func main() {
 		})
 	}
 
-	g.Go(func() error {
-		logrus.Infof("server listening on address %s", spec.Address)
-		return http.ListenAndServe(spec.Address, nil)
-	})
+	if spec.UnixSocket != "" {
+		g.Go(func() error {
+			l, err := net.Listen("unix", spec.UnixSocket)
+			if err != nil {
+				return err
+			}
+			logrus.Infof("server listening on socket %s", spec.UnixSocket)
+			return http.Serve(l, nil)
+		})
+	} else {
+		g.Go(func() error {
+			logrus.Infof("server listening on address %s", spec.Address)
+			return http.ListenAndServe(spec.Address, nil)
+		})
+	}
 
 	if err := g.Wait(); err != nil {
 		logrus.Fatal(err)
