Index: src/engine/ftp/rawtransfer.cpp
===================================================================
--- a/src/engine/ftp/rawtransfer.cpp	(revision 11406)
+++ b/src/engine/ftp/rawtransfer.cpp	(working copy)
@@ -399,7 +399,11 @@
 	}
 
 	std::wstring const peerIP = fz::to_wstring(controlSocket_.socket_->peer_ip());
-	if (!fz::is_routable_address(host_) && fz::is_routable_address(peerIP)) {
+	std::wstring const zeroIP = fz::to_wstring(std::string("0.0.0.0"));
+	if (
+		std::wcscmp(zeroIP.c_str(), peerIP.c_str())
+		|| (!fz::is_routable_address(host_) && fz::is_routable_address(peerIP))
+	) {
 		if (options_.get_int(OPTION_PASVREPLYFALLBACKMODE) != 1 || bTriedActive) {
 			log(logmsg::status, _("Server sent passive reply with unroutable address. Using server address instead."));
 			log(logmsg::debug_info, L"  Reply: %s, peer: %s", host_, peerIP);
