diff --git a/src/lib.rs b/src/lib.rs
index 86cf400b6..79abfdbbd 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -93,8 +93,6 @@ pub async fn start_lemmy_server() -> Result<(), LemmyError> {
     println!("federation enabled, host is {}", &settings.hostname);
   }
 
-  check_private_instance_and_federation_enabled(&local_site)?;
-
   // Set up the rate limiter
   let rate_limit_config =
     local_site_rate_limit_to_rate_limit_config(&site_view.local_site_rate_limit);
