diff --git a/gamecheckin.py b/gamecheckin.py
index 1e810ac..0328b6c 100644
--- a/gamecheckin.py
+++ b/gamecheckin.py
@@ -153,7 +153,7 @@ class GameCheckin:
                             f"{tools.get_item(self.checkin_rewards[sign_days])}")
                     else:
                         s = "账号签到失败！"
-                        if data["data"] != "" and data.get("data").get("success", -1):
+                        if data.get("data") and data.get("data").get("success", -1):
                             s += "原因: 验证码\njson信息:" + req.text
                         log.warning(s)
                         return_data += f"\n{account[0]}，触发验证码，本次签到失败"
