diff --git a/src/firewall/nft.rs b/src/firewall/nft.rs
index 3d96e7b8..c3156999 100644
--- a/src/firewall/nft.rs
+++ b/src/firewall/nft.rs
@@ -224,34 +224,6 @@ impl firewall::FirewallDriver for Nftables {
             ));
         }
 
-        // Forward chain: ct state invalid drop
-        let match_deny = |r: &schema::Rule| -> bool {
-            for statement in r.expr.deref() {
-                match statement {
-                    stmt::Statement::Drop(_) => return true,
-                    _ => continue,
-                }
-            }
-            false
-        };
-        if get_matching_rules_in_chain(&existing_rules, FORWARDCHAIN, match_deny).is_empty() {
-            batch.add(make_rule(
-                Cow::Borrowed(FORWARDCHAIN),
-                Cow::Borrowed(&[
-                    stmt::Statement::Match(stmt::Match {
-                        left: expr::Expression::Named(expr::NamedExpression::CT(expr::CT {
-                            key: Cow::Borrowed("state"),
-                            family: None,
-                            dir: None,
-                        })),
-                        right: expr::Expression::String(Cow::Borrowed("invalid")),
-                        op: stmt::Operator::IN,
-                    }),
-                    stmt::Statement::Drop(None),
-                ]),
-            ));
-        }
-
         // Forward chain: jump NETAVARK-ISOLATION-1
         if get_matching_rules_in_chain(
             &existing_rules,
