diff --git a/src/input/mod.rs b/src/input/mod.rs
index 50567530..3a134cd9 100644
--- a/src/input/mod.rs
+++ b/src/input/mod.rs
@@ -3839,13 +3839,13 @@ impl State {
         }
 
         if event.fingers() == 3 {
-            self.niri.gesture_swipe_3f_cumulative = Some((0., 0.));
+            self.niri.layout.overview_gesture_begin();
+            self.niri.queue_redraw_all();
 
             // We handled this event.
             return;
         } else if event.fingers() == 4 {
-            self.niri.layout.overview_gesture_begin();
-            self.niri.queue_redraw_all();
+            self.niri.gesture_swipe_3f_cumulative = Some((0., 0.));
 
             // We handled this event.
             return;
