diff --git a/lib/cli.zsh b/lib/cli.zsh
index fed00d21..94350a3d 100644
--- a/lib/cli.zsh
+++ b/lib/cli.zsh
@@ -471,7 +471,7 @@ function _omz::plugin::load {
   # Note that we still have to pass -d "$_comp_dumpfile", so that compinit
   # doesn't use the default zcompdump location (${ZDOTDIR:-$HOME}/.zcompdump).
   if (( has_completion )); then
-    compinit -D -d "$_comp_dumpfile"
+    compinit -D
   fi
 }
 
diff --git a/oh-my-zsh.sh b/oh-my-zsh.sh
index e94c2f41..efee2667 100644
--- a/oh-my-zsh.sh
+++ b/oh-my-zsh.sh
@@ -107,46 +107,7 @@ if [[ -z "$ZSH_COMPDUMP" ]]; then
   ZSH_COMPDUMP="${ZDOTDIR:-$HOME}/.zcompdump-${SHORT_HOST}-${ZSH_VERSION}"
 fi
 
-# Construct zcompdump OMZ metadata
-zcompdump_revision="#omz revision: $(builtin cd -q "$ZSH"; git rev-parse HEAD 2>/dev/null)"
-zcompdump_fpath="#omz fpath: $fpath"
-
-# Delete the zcompdump file if OMZ zcompdump metadata changed
-if ! command grep -q -Fx "$zcompdump_revision" "$ZSH_COMPDUMP" 2>/dev/null \
-   || ! command grep -q -Fx "$zcompdump_fpath" "$ZSH_COMPDUMP" 2>/dev/null; then
-  command rm -f "$ZSH_COMPDUMP"
-  zcompdump_refresh=1
-fi
-
-if [[ "$ZSH_DISABLE_COMPFIX" != true ]]; then
-  source "$ZSH/lib/compfix.zsh"
-  # Load only from secure directories
-  compinit -i -d "$ZSH_COMPDUMP"
-  # If completion insecurities exist, warn the user
-  handle_completion_insecurities &|
-else
-  # If the user wants it, load from all found directories
-  compinit -u -d "$ZSH_COMPDUMP"
-fi
-
-# Append zcompdump metadata if missing
-if (( $zcompdump_refresh )) \
-  || ! command grep -q -Fx "$zcompdump_revision" "$ZSH_COMPDUMP" 2>/dev/null; then
-  # Use `tee` in case the $ZSH_COMPDUMP filename is invalid, to silence the error
-  # See https://github.com/ohmyzsh/ohmyzsh/commit/dd1a7269#commitcomment-39003489
-  tee -a "$ZSH_COMPDUMP" &>/dev/null <<EOF
-
-$zcompdump_revision
-$zcompdump_fpath
-EOF
-fi
-unset zcompdump_revision zcompdump_fpath zcompdump_refresh
-
-# zcompile the completion dump file if the .zwc is older or missing.
-if command mkdir "${ZSH_COMPDUMP}.lock" 2>/dev/null; then
-  zrecompile -q -p "$ZSH_COMPDUMP"
-  command rm -rf "$ZSH_COMPDUMP.zwc.old" "${ZSH_COMPDUMP}.lock" 
-fi
+compinit -D
 
 # Load all of the config files in ~/oh-my-zsh that end in .zsh
 # TIP: Add files you don't want in git to .gitignore
