diff --git a/app/server.py b/app/server.py
index 7201666..c78c5cf 100644
--- a/app/server.py
+++ b/app/server.py
@@ -15,6 +15,7 @@ app = Flask(__name__)
 load_dotenv()
 
 API_KEY = os.getenv('API_KEY', DEFAULT_CONFIGS["API_KEY"])
+HOST = os.getenv('HOST', '0.0.0.0')
 PORT = int(os.getenv('PORT', str(DEFAULT_CONFIGS["PORT"])))
 
 DEFAULT_VOICE = os.getenv('DEFAULT_VOICE', DEFAULT_CONFIGS["DEFAULT_VOICE"])
@@ -173,5 +174,5 @@ print(f" * TTS Endpoint: http://localhost:{PORT}/v1/audio/speech")
 print(f" ")
 
 if __name__ == '__main__':
-    http_server = WSGIServer(('0.0.0.0', PORT), app)
+    http_server = WSGIServer((HOST, PORT), app)
     http_server.serve_forever()
