diff --git a/src/modules/module-roc-sink.c b/src/modules/module-roc-sink.c
index 06af90f47..50ede67fd 100644
--- a/src/modules/module-roc-sink.c
+++ b/src/modules/module-roc-sink.c
@@ -265,6 +265,7 @@ static int roc_sink_setup(struct module_roc_sink_data *data)
 	sender_config.frame_encoding.channels = ROC_CHANNEL_LAYOUT_STEREO;
 	sender_config.frame_encoding.format = ROC_FORMAT_PCM_FLOAT32;
 	sender_config.fec_encoding = data->fec_code;
+	sender_config.packet_encoding = ROC_PACKET_ENCODING_AVP_L16_STEREO;
 
 	info.rate = data->rate;
 
