diff --git a/plasmoid/contents/ui/main.qml b/plasmoid/contents/ui/main.qml
index 5a6a651..d8408ac 100644
--- a/plasmoid/contents/ui/main.qml
+++ b/plasmoid/contents/ui/main.qml
@@ -19,10 +19,21 @@ PlasmoidItem {
             Layout.minimumWidth = text.contentWidth
         }
 
+        Timer {
+            id: timer
+        }
+
+        function delay(delayTime,cb) {
+            timer.interval = delayTime;
+            timer.repeat = false;
+            timer.triggered.connect(cb);
+            timer.start();
+        }
+
         WebSocket {
             id: socket
             url: "ws://127.0.0.1:15648/ws"
-            onTextMessageReceived: {
+            onTextMessageReceived: function(message) {
                 var lyric = JSON.parse(message).lyric || ""
                 if (lyric.length > plasmoid.configuration.characterLimit) {
                     lyric = lyric.slice(0, plasmoid.configuration.characterLimit) + "..."
@@ -31,6 +42,18 @@ PlasmoidItem {
                 updateLayoutSize()
                 return
             }
+            onStatusChanged: function(status) {
+                if (status == WebSocket.Closed || status == WebSocket.Error) {
+                    text.text = "[" + i18n("Reconnecting") + "...]"
+                    updateLayoutSize()
+                    socket.active = false
+                    delay(500, function() {
+                        if (socket.active == false) {
+                            socket.active = true
+                        }
+                    })
+                }
+            }
             active: true
         }
 
