diff --git a/src/base/bittorrent/sessionimpl.cpp b/src/base/bittorrent/sessionimpl.cpp
index 1866b7157..a4513c479 100644
--- a/src/base/bittorrent/sessionimpl.cpp
+++ b/src/base/bittorrent/sessionimpl.cpp
@@ -2804,7 +2804,7 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
     if (m_loadingTorrents.contains(id) || (infoHash.isHybrid() && m_loadingTorrents.contains(altID)))
     {
         emit addTorrentFailed(infoHash, {AddTorrentError::DuplicateTorrent, tr("Duplicate torrent")});
-        return false;
+        return true;
     }
 
     if (Torrent *torrent = findTorrent(infoHash))
@@ -2823,7 +2823,7 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
             LogMsg(tr("Detected an attempt to add a duplicate torrent. Existing torrent: %1. Result: %2")
                     .arg(torrent->name(), message));
             emit addTorrentFailed(infoHash, {AddTorrentError::DuplicateTorrent, message});
-            return false;
+            return true;
         }
 
         const bool isPrivate = torrent->isPrivate() || (hasMetadata && source.info()->isPrivate());
@@ -2833,7 +2833,7 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
             LogMsg(tr("Detected an attempt to add a duplicate torrent. Existing torrent: %1. Result: %2")
                     .arg(torrent->name(), message));
             emit addTorrentFailed(infoHash, {AddTorrentError::DuplicateTorrent, message});
-            return false;
+            return true;
         }
 
         // merge trackers and web seeds
@@ -2844,7 +2844,7 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
         LogMsg(tr("Detected an attempt to add a duplicate torrent. Existing torrent: %1. Result: %2")
                 .arg(torrent->name(), message));
         emit addTorrentFailed(infoHash, {AddTorrentError::DuplicateTorrent, message});
-        return false;
+        return true;
     }
 
     // It looks illogical that we don't just use an existing handle,
