diff --git a/src/base/bittorrent/sessionimpl.cpp b/src/base/bittorrent/sessionimpl.cpp
index b8958c589..f3163d2df 100644
--- a/src/base/bittorrent/sessionimpl.cpp
+++ b/src/base/bittorrent/sessionimpl.cpp
@@ -2760,7 +2760,7 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
             LogMsg(tr("Detected an attempt to add a duplicate torrent. Existing torrent: \"%1\". Torrent infohash: %2. Result: %3")
                     .arg(torrent->name(), torrent->infoHash().toString(), message));
             emit addTorrentFailed(infoHash, {AddTorrentError::DuplicateTorrent, message});
-            return false;
+            return true;
         }
 
         const bool isPrivate = torrent->isPrivate() || (hasMetadata && source.info()->isPrivate());
@@ -2770,7 +2770,7 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
             LogMsg(tr("Detected an attempt to add a duplicate torrent. Existing torrent: \"%1\". Torrent infohash: %2. Result: %3")
                     .arg(torrent->name(), torrent->infoHash().toString(), message));
             emit addTorrentFailed(infoHash, {AddTorrentError::DuplicateTorrent, message});
-            return false;
+            return true;
         }
 
         // merge trackers and web seeds
@@ -2781,7 +2781,7 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
         LogMsg(tr("Detected an attempt to add a duplicate torrent. Existing torrent: \"%1\". Torrent infohash: %2. Result: %3")
                 .arg(torrent->name(), torrent->infoHash().toString(), message));
         emit addTorrentFailed(infoHash, {AddTorrentError::DuplicateTorrent, message});
-        return false;
+        return true;
     }
 
     // It looks illogical that we don't just use an existing handle,
