diff --git a/include/ZeroTierOne.h b/include/ZeroTierOne.h
index 448234693..d4a6db699 100644
--- a/include/ZeroTierOne.h
+++ b/include/ZeroTierOne.h
@@ -128,7 +128,7 @@ extern "C" {
 /**
  * Maximum number of pushed routes on a network
  */
-#define ZT_MAX_NETWORK_ROUTES 128
+#define ZT_MAX_NETWORK_ROUTES 1024
 
 /**
  * Maximum number of statically assigned IP addresses per network endpoint using ZT address management (not DHCP)
