diff --git a/pkgs/os-specific/linux/firmware/raspberrypi/default.nix b/pkgs/os-specific/linux/firmware/raspberrypi/default.nix
index ab60a10447ef6..497bc3c326b2a 100644
--- a/pkgs/os-specific/linux/firmware/raspberrypi/default.nix
+++ b/pkgs/os-specific/linux/firmware/raspberrypi/default.nix
@@ -3,13 +3,13 @@
 stdenvNoCC.mkDerivation rec {
   # NOTE: this should be updated with linux_rpi
   pname = "raspberrypi-firmware";
-  version = "1.20240926";
+  version = "1.20241008";
 
   src = fetchFromGitHub {
     owner = "raspberrypi";
     repo = "firmware";
     rev = version;
-    hash = "sha256-MCutxzdSFoZ4hn2Fxk2AHHgWCt/Jgc+reqJZHUuSKOc=";
+    hash = "sha256-4gnK0KbqFnjBmWia9Jt2gveVWftmHrprpwBqYVqE/k0=";
   };
 
   installPhase = ''
diff --git a/pkgs/os-specific/linux/kernel/linux-rpi.nix b/pkgs/os-specific/linux/kernel/linux-rpi.nix
index 54c5ace091542..4c701e5556a99 100644
--- a/pkgs/os-specific/linux/kernel/linux-rpi.nix
+++ b/pkgs/os-specific/linux/kernel/linux-rpi.nix
@@ -2,8 +2,8 @@
 
 let
   # NOTE: raspberrypifw & raspberryPiWirelessFirmware should be updated with this
-  modDirVersion = "6.6.31";
-  tag = "stable_20240529";
+  modDirVersion = "6.6.51";
+  tag = "stable_20241008";
 in
 lib.overrideDerivation (buildLinux (args // {
   version = "${modDirVersion}-${tag}";
@@ -14,7 +14,7 @@ lib.overrideDerivation (buildLinux (args // {
     owner = "raspberrypi";
     repo = "linux";
     rev = tag;
-    hash = "sha256-UWUTeCpEN7dlFSQjog6S3HyEWCCnaqiUqV5KxCjYink=";
+    hash = "sha256-phCxkuO+jUGZkfzSrBq6yErQeO2Td+inIGHxctXbD5U=";
   };
 
   defconfig = {
@@ -24,37 +24,10 @@ lib.overrideDerivation (buildLinux (args // {
     "4" = "bcm2711_defconfig";
   }.${toString rpiVersion};
 
-  structuredExtraConfig = (args.structuredExtraConfig or {}) // (with lib.kernel; {
-    # Workaround https://github.com/raspberrypi/linux/issues/6198
-    # Needed because NixOS 24.05+ sets DRM_SIMPLEDRM=y which pulls in
-    # DRM_KMS_HELPER=y.
-    BACKLIGHT_CLASS_DEVICE = yes;
-  });
-
   features = {
     efiBootStub = false;
   } // (args.features or {});
 
-  kernelPatches = (args.kernelPatches or []) ++ [
-    # Fix compilation errors due to incomplete patch backport.
-    # https://github.com/raspberrypi/linux/pull/6223
-    {
-      name = "gpio-pwm_-_pwm_apply_might_sleep.patch";
-      patch = fetchpatch {
-        url = "https://github.com/peat-psuwit/rpi-linux/commit/879f34b88c60dd59765caa30576cb5bfb8e73c56.patch";
-        hash = "sha256-HlOkM9EFmlzOebCGoj7lNV5hc0wMjhaBFFZvaRCI0lI=";
-      };
-    }
-
-    {
-      name = "ir-rx51_-_pwm_apply_might_sleep.patch";
-      patch = fetchpatch {
-        url = "https://github.com/peat-psuwit/rpi-linux/commit/23431052d2dce8084b72e399fce82b05d86b847f.patch";
-        hash = "sha256-UDX/BJCJG0WVndP/6PbPK+AZsfU3vVxDCrpn1kb1kqE=";
-      };
-    }
-  ];
-
   extraMeta = if (rpiVersion < 3) then {
     platforms = with lib.platforms; arm;
     hydraPlatforms = [];
