diff --git a/pkgs/os-specific/linux/nvidia-x11/persistenced.nix b/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
index d125cf5204478f..cf9a8f0f404b62 100644
--- a/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
@@ -5,7 +5,9 @@ nvidia_x11: sha256:
   lib,
   fetchFromGitHub,
   m4,
+  glibc,
   libtirpc,
+  pkg-config,
 }:
 
 stdenv.mkDerivation {
@@ -19,8 +21,21 @@ stdenv.mkDerivation {
     inherit sha256;
   };
 
-  nativeBuildInputs = [ m4 ];
-  buildInputs = [ libtirpc ];
+  env = {
+    LIBRARY_PATH = "${glibc}/lib";
+    NIX_CFLAGS_COMPILE = toString [ "-I${libtirpc.dev}/include/tirpc" ];
+  };
+  NIX_LDFLAGS = [ "-ltirpc" ];
+
+  nativeBuildInputs = [
+    m4
+    pkg-config
+  ];
+
+  buildInputs = [
+    libtirpc
+    stdenv.cc.cc.lib
+  ];
 
   makeFlags = nvidia_x11.makeFlags ++ [ "DATE=true" ];
 
@@ -36,13 +51,10 @@ stdenv.mkDerivation {
       $out/bin/nvidia-persistenced
   '';
 
-  env.NIX_CFLAGS_COMPILE = toString [ "-I${libtirpc.dev}/include/tirpc" ];
-  NIX_LDFLAGS = [ "-ltirpc" ];
-
-  meta = with lib; {
-    homepage = "https://www.nvidia.com/object/unix.html";
-    description = "Settings application for NVIDIA graphics cards";
-    license = licenses.unfreeRedistributable;
+  meta = {
+    homepage = "https://github.com/NVIDIA/nvidia-persistenced";
+    description = "NVIDIA driver persistence daemon";
+    license = lib.licenses.mit;
     platforms = nvidia_x11.meta.platforms;
     maintainers = [ ];
     mainProgram = "nvidia-persistenced";
