diff --git a/pkgs/by-name/az/azure-cli/package.nix b/pkgs/by-name/az/azure-cli/package.nix
index 2390004e2b073c..8bd266429572fb 100644
--- a/pkgs/by-name/az/azure-cli/package.nix
+++ b/pkgs/by-name/az/azure-cli/package.nix
@@ -26,14 +26,14 @@
 }:
 
 let
-  version = "2.75.0";
+  version = "2.77.0";
 
   src = fetchFromGitHub {
     name = "azure-cli-${version}-src";
     owner = "Azure";
     repo = "azure-cli";
     tag = "azure-cli-${version}";
-    hash = "sha256-u6umAqRUfiACt23mxTtfosLdxKSPvDVJMkVjPCtxr24=";
+    hash = "sha256-+AvfZIeGxQ+27gpg1y3fMMN7fBP4ATXr7By+dJpux3k=";
   };
 
   # put packages that needs to be overridden in the py package scope
@@ -154,7 +154,7 @@ py.pkgs.toPythonApplication (
     ];
 
     # Dependencies from:
-    # https://github.com/Azure/azure-cli/blob/azure-cli-2.62.0/src/azure-cli/setup.py#L52
+    # https://github.com/Azure/azure-cli/blob/azure-cli-2.77.0/src/azure-cli/setup.py#L52
     # Please, keep ordered by upstream file order. It facilitates reviews.
     propagatedBuildInputs =
       with py.pkgs;
@@ -166,7 +166,6 @@ py.pkgs.toPythonApplication (
         azure-cosmos
         azure-data-tables
         azure-datalake-store
-        azure-graphrbac
         azure-keyvault-administration
         azure-keyvault-certificates
         azure-keyvault-keys
@@ -189,10 +188,8 @@ py.pkgs.toPythonApplication (
         azure-mgmt-containerregistry
         azure-mgmt-containerservice
         azure-mgmt-cosmosdb
-        azure-mgmt-databoxedge
         azure-mgmt-datalake-store
         azure-mgmt-datamigration
-        azure-mgmt-dns
         azure-mgmt-eventgrid
         azure-mgmt-eventhub
         azure-mgmt-extendedlocation
@@ -203,24 +200,29 @@ py.pkgs.toPythonApplication (
         azure-mgmt-iothubprovisioningservices
         azure-mgmt-keyvault
         azure-mgmt-loganalytics
-        azure-mgmt-managedservices
         azure-mgmt-managementgroups
         azure-mgmt-maps
         azure-mgmt-marketplaceordering
         azure-mgmt-media
         azure-mgmt-monitor
         azure-mgmt-msi
-        azure-mgmt-mysqlflexibleservers
         azure-mgmt-netapp
         azure-mgmt-policyinsights
         azure-mgmt-postgresqlflexibleservers
         azure-mgmt-privatedns
         azure-mgmt-rdbms
+        azure-mgmt-mysqlflexibleservers
         azure-mgmt-recoveryservicesbackup
         azure-mgmt-recoveryservices
-        azure-mgmt-redis
         azure-mgmt-redhatopenshift
-        azure-mgmt-resource
+        azure-mgmt-redis
+        azure-mgmt-resource-all
+        # Added through azure-mgmt-resource-all package
+        # azure-mgmt-resource
+        # azure-mgmt-resource-deployments
+        # azure-mgmt-resource-deploymentscripts
+        # azure-mgmt-resource-deploymentstacks
+        # azure-mgmt-resource-templatespecs
         azure-mgmt-search
         azure-mgmt-security
         azure-mgmt-servicebus
@@ -250,6 +252,7 @@ py.pkgs.toPythonApplication (
         javaproperties
         jsondiff
         packaging
+        paramiko
         pycomposefile
         pygithub
         pynacl
@@ -324,7 +327,6 @@ py.pkgs.toPythonApplication (
       "azure.cli.telemetry"
       "azure.cosmos"
       "azure.datalake.store"
-      "azure.graphrbac"
       "azure.keyvault"
       "azure.mgmt.advisor"
       "azure.mgmt.apimanagement"
@@ -344,7 +346,6 @@ py.pkgs.toPythonApplication (
       "azure.mgmt.containerservice"
       "azure.mgmt.cosmosdb"
       "azure.mgmt.datamigration"
-      "azure.mgmt.dns"
       "azure.mgmt.eventgrid"
       "azure.mgmt.eventhub"
       "azure.mgmt.hdinsight"
@@ -354,7 +355,6 @@ py.pkgs.toPythonApplication (
       "azure.mgmt.iothubprovisioningservices"
       "azure.mgmt.keyvault"
       "azure.mgmt.loganalytics"
-      "azure.mgmt.managedservices"
       "azure.mgmt.managementgroups"
       "azure.mgmt.maps"
       "azure.mgmt.marketplaceordering"
@@ -369,6 +369,8 @@ py.pkgs.toPythonApplication (
       "azure.mgmt.recoveryservicesbackup"
       "azure.mgmt.redis"
       "azure.mgmt.resource"
+      "azure.mgmt.resource.deployments"
+      "azure.mgmt.resource.deployments.models"
       "azure.mgmt.search"
       "azure.mgmt.security"
       "azure.mgmt.servicebus"
diff --git a/pkgs/by-name/az/azure-cli/python-packages.nix b/pkgs/by-name/az/azure-cli/python-packages.nix
index 13ab4f80d9107c..819a186f2d2098 100644
--- a/pkgs/by-name/az/azure-cli/python-packages.nix
+++ b/pkgs/by-name/az/azure-cli/python-packages.nix
@@ -1,4 +1,5 @@
 {
+  lib,
   stdenv,
   python3,
   fetchPypi,
@@ -191,6 +192,41 @@ let
         overrideAzureMgmtPackage super.azure-mgmt-redhatopenshift "1.5.0" "tar.gz"
           "sha256-Uft0KcOciKzJ+ic9n4nxkwNSBmKZam19jhEiqY9fJSc=";
 
+      # azure.mgmt.resource will shadow the other azure.mgmt.resource.* packages unless we merge them together
+      azure-mgmt-resource-all = py.pkgs.buildPythonPackage {
+        pname = "azure-mgmt-resource-all";
+        inherit version;
+
+        format = "other"; # we're not building from sdist/wheel
+
+        src = py.pkgs.azure-mgmt-resource.src;
+
+        # No real build, just symlink all site-packages into one dir
+        installPhase = ''
+          runHook preInstall
+
+          mkdir -p $out/${py.sitePackages}
+          for pkg in ${
+            lib.concatStringsSep " " (
+              map (p: "${p}") [
+                py.pkgs.azure-mgmt-resource
+                py.pkgs.azure-mgmt-resource-deployments
+                py.pkgs.azure-mgmt-resource-deploymentscripts
+                py.pkgs.azure-mgmt-resource-deploymentstacks
+                py.pkgs.azure-mgmt-resource-templatespecs
+              ]
+            )
+          }; do
+            # Copy recursively, keep symlinks, skip duplicates silently
+            cp -rs --no-preserve=mode "$pkg/${py.sitePackages}/." "$out/${py.sitePackages}/" || true
+          done
+
+          runHook postInstall
+        '';
+
+        doCheck = false;
+      };
+
       # ImportError: cannot import name 'IPRule' from 'azure.mgmt.signalr.models'
       azure-mgmt-signalr =
         overrideAzureMgmtPackage super.azure-mgmt-signalr "2.0.0b2" "tar.gz"
diff --git a/pkgs/development/python-modules/azure-mgmt-resource-deployments/default.nix b/pkgs/development/python-modules/azure-mgmt-resource-deployments/default.nix
new file mode 100644
index 00000000000000..5eb6019ea8390e
--- /dev/null
+++ b/pkgs/development/python-modules/azure-mgmt-resource-deployments/default.nix
@@ -0,0 +1,55 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  azure-mgmt-core,
+  azure-common,
+  isodate,
+  pythonOlder,
+  setuptools,
+  typing-extensions,
+  azure-cli,
+}:
+
+buildPythonPackage rec {
+  pname = "azure-mgmt-resource-deployments";
+  version = "1.0.0b1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.9";
+
+  src = fetchPypi {
+    pname = "azure_mgmt_resource_deployments";
+    inherit version;
+    hash = "sha256-c1m0JliCbn5/8T5tuwxJDpX8yV28oiTSuFz3GtdTXx0=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    azure-common
+    azure-mgmt-core
+    isodate
+    typing-extensions
+  ];
+
+  # Module has no tests
+  doCheck = false;
+
+  pythonNamespaces = [
+    "azure.mgmt.resource.deployments"
+  ];
+
+  pythonImportsCheck = [
+    "azure.mgmt.resource.deployments"
+    "azure.mgmt.resource.deployments.models"
+  ];
+
+  meta = with lib; {
+    description = "Microsoft Azure SDK for Python";
+    homepage = "https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/resources/azure-mgmt-resource-deployments";
+    changelog = "https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-resource-deployments_${version}/sdk/resources/azure-mgmt-resource-deployments/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = azure-cli.meta.maintainers;
+  };
+}
diff --git a/pkgs/development/python-modules/azure-mgmt-resource-deploymentscripts/default.nix b/pkgs/development/python-modules/azure-mgmt-resource-deploymentscripts/default.nix
new file mode 100644
index 00000000000000..3fbca5fec29b04
--- /dev/null
+++ b/pkgs/development/python-modules/azure-mgmt-resource-deploymentscripts/default.nix
@@ -0,0 +1,54 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  azure-mgmt-core,
+  azure-common,
+  isodate,
+  pythonOlder,
+  setuptools,
+  typing-extensions,
+  azure-cli,
+}:
+
+buildPythonPackage rec {
+  pname = "azure-mgmt-resource-deploymentscripts";
+  version = "1.0.0b1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.9";
+
+  src = fetchPypi {
+    pname = "azure_mgmt_resource_deploymentscripts";
+    inherit version;
+    hash = "sha256-Vm2FWVPpSbsrNMtD4ecwVKqnkoHHRhO3Rf/duCyAI3U=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    azure-common
+    azure-mgmt-core
+    isodate
+    typing-extensions
+  ];
+
+  # Module has no tests
+  doCheck = false;
+
+  pythonNamespaces = [
+    "azure.mgmt.resource.deploymentscripts"
+  ];
+
+  pythonImportsCheck = [
+    "azure.mgmt.resource.deploymentscripts"
+  ];
+
+  meta = with lib; {
+    description = "Microsoft Azure SDK for Python";
+    homepage = "https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/resources/azure-mgmt-resource-deploymentscripts";
+    changelog = "https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-resource-deploymentscripts_${version}/sdk/resources/azure-mgmt-resource-deploymentscripts/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = azure-cli.meta.maintainers;
+  };
+}
diff --git a/pkgs/development/python-modules/azure-mgmt-resource-deploymentstacks/default.nix b/pkgs/development/python-modules/azure-mgmt-resource-deploymentstacks/default.nix
new file mode 100644
index 00000000000000..7740e636ba4b92
--- /dev/null
+++ b/pkgs/development/python-modules/azure-mgmt-resource-deploymentstacks/default.nix
@@ -0,0 +1,54 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  azure-mgmt-core,
+  azure-common,
+  isodate,
+  pythonOlder,
+  setuptools,
+  typing-extensions,
+  azure-cli,
+}:
+
+buildPythonPackage rec {
+  pname = "azure-mgmt-resource-deploymentstacks";
+  version = "1.0.0b1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.9";
+
+  src = fetchPypi {
+    pname = "azure_mgmt_resource_deploymentstacks";
+    inherit version;
+    hash = "sha256-Sbh25FwPW6uI7EfXxW6Ps0+9/P5bgxo6KiHoF/PZcy4=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    azure-common
+    azure-mgmt-core
+    isodate
+    typing-extensions
+  ];
+
+  # Module has no tests
+  doCheck = false;
+
+  pythonNamespaces = [
+    "azure.mgmt.resource.deploymentstacks"
+  ];
+
+  pythonImportsCheck = [
+    "azure.mgmt.resource.deploymentstacks"
+  ];
+
+  meta = with lib; {
+    description = "Microsoft Azure SDK for Python";
+    homepage = "https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/resources/azure-mgmt-resource-deploymentstacks";
+    changelog = "https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-resource-deploymentstacks_${version}/sdk/resources/azure-mgmt-resource-deploymentstacks/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = azure-cli.meta.maintainers;
+  };
+}
diff --git a/pkgs/development/python-modules/azure-mgmt-resource-templatespecs/default.nix b/pkgs/development/python-modules/azure-mgmt-resource-templatespecs/default.nix
new file mode 100644
index 00000000000000..75c261f3d5f8bf
--- /dev/null
+++ b/pkgs/development/python-modules/azure-mgmt-resource-templatespecs/default.nix
@@ -0,0 +1,54 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  azure-mgmt-core,
+  azure-common,
+  isodate,
+  pythonOlder,
+  setuptools,
+  typing-extensions,
+  azure-cli,
+}:
+
+buildPythonPackage rec {
+  pname = "azure-mgmt-resource-templatespecs";
+  version = "1.0.0b1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.9";
+
+  src = fetchPypi {
+    pname = "azure_mgmt_resource_templatespecs";
+    inherit version;
+    hash = "sha256-D55zmrQ9sq2HDq5d8bXEv6BQC76hxuWKpeLpw4X6y8U=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    azure-common
+    azure-mgmt-core
+    isodate
+    typing-extensions
+  ];
+
+  # Module has no tests
+  doCheck = false;
+
+  pythonNamespaces = [
+    "azure.mgmt.resource.templatespecs"
+  ];
+
+  pythonImportsCheck = [
+    "azure.mgmt.resource.templatespecs"
+  ];
+
+  meta = with lib; {
+    description = "Microsoft Azure SDK for Python";
+    homepage = "https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/resources/azure-mgmt-resource-templatespecs";
+    changelog = "https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-resource-templatespecs_${version}/sdk/resources/azure-mgmt-resource-templatespecs/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = azure-cli.meta.maintainers;
+  };
+}
diff --git a/pkgs/top-level/python-packages.nix b/pkgs/top-level/python-packages.nix
index 6b3380f86057c1..de4a065fc7e780 100644
--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -1528,6 +1528,22 @@ self: super: with self; {
 
   azure-mgmt-resource = callPackage ../development/python-modules/azure-mgmt-resource { };
 
+  azure-mgmt-resource-deployments =
+    callPackage ../development/python-modules/azure-mgmt-resource-deployments
+      { };
+
+  azure-mgmt-resource-deploymentscripts =
+    callPackage ../development/python-modules/azure-mgmt-resource-deploymentscripts
+      { };
+
+  azure-mgmt-resource-deploymentstacks =
+    callPackage ../development/python-modules/azure-mgmt-resource-deploymentstacks
+      { };
+
+  azure-mgmt-resource-templatespecs =
+    callPackage ../development/python-modules/azure-mgmt-resource-templatespecs
+      { };
+
   azure-mgmt-scheduler = callPackage ../development/python-modules/azure-mgmt-scheduler { };
 
   azure-mgmt-search = callPackage ../development/python-modules/azure-mgmt-search { };
