diff --git a/pkgs/by-name/pa/pagmo2/package.nix b/pkgs/by-name/pa/pagmo2/package.nix
index 04dd5b35123cf..48e4c3247e63e 100644
--- a/pkgs/by-name/pa/pagmo2/package.nix
+++ b/pkgs/by-name/pa/pagmo2/package.nix
@@ -22,11 +22,6 @@ stdenv.mkDerivation (finalAttrs: {
     sha256 = "sha256-ido3e0hQLDEPT0AmsfAVTPlGbWe5QBkxgRO6Fg1wp/c=";
   };
 
-  outputs = [
-    "out"
-    "dev"
-  ];
-
   nativeBuildInputs = [ cmake ];
 
   buildInputs = [
