diff --git a/pkgs/by-name/ac/acme-sh/package.nix b/pkgs/by-name/ac/acme-sh/package.nix
index c79bcebda086..762d48e5c07f 100644
--- a/pkgs/by-name/ac/acme-sh/package.nix
+++ b/pkgs/by-name/ac/acme-sh/package.nix
@@ -44,7 +44,7 @@ stdenv.mkDerivation rec {
     runHook preInstall
 
     mkdir -p $out $out/bin $out/libexec
-    cp -R $src/* $_
+    cp -R * $_
     makeWrapper $out/libexec/acme.sh $out/bin/acme.sh \
       --prefix PATH : "${binPath}"
 
