diff --git a/nixos/modules/services/misc/bepasty.nix b/nixos/modules/services/misc/bepasty.nix
index d890fe507ae7..107f0426af8e 100644
--- a/nixos/modules/services/misc/bepasty.nix
+++ b/nixos/modules/services/misc/bepasty.nix
@@ -76,6 +76,14 @@ in
                   '';
                 };
 
+                extraConfigFile = lib.mkOption {
+                  type = lib.types.nullOr lib.types.str;
+                  description = ''
+                    Path of file to be included as extra configuration.
+                    '';
+                  default = null;
+                };
+
                 secretKey = lib.mkOption {
                   type = lib.types.str;
                   description = ''
@@ -174,6 +182,7 @@ in
               SECRET_KEY="$(cat "${server.secretKeyFile}")"
               DEFAULT_PERMISSIONS="${server.defaultPermissions}"
               ${server.extraConfig}
+              $(cat "${server.extraConfigFile}")
               EOF
             '';
           ExecStart = ''
