diff --git a/nixos/modules/services/misc/bepasty.nix b/nixos/modules/services/misc/bepasty.nix
index 76e85a91317c..c688780638e7 100644
--- a/nixos/modules/services/misc/bepasty.nix
+++ b/nixos/modules/services/misc/bepasty.nix
@@ -66,6 +66,14 @@ in
               '';
           };
 
+          extraConfigFile = lib.mkOption {
+            type = lib.types.nullOr lib.types.str;
+            description = ''
+              Path of file to be included as extra configuration.
+              '';
+            default = null;
+          };
+
           secretKey = lib.mkOption {
             type = lib.types.str;
             description = ''
@@ -152,6 +160,7 @@ in
               SECRET_KEY="$(cat "${server.secretKeyFile}")"
               DEFAULT_PERMISSIONS="${server.defaultPermissions}"
               ${server.extraConfig}
+              $(cat "${server.extraConfigFile}")
               EOF
             '';
             ExecStart = ''${gunicorn}/bin/gunicorn bepasty.wsgi --name ${name} \
