diff --git a/nixos/modules/services/misc/bepasty.nix b/nixos/modules/services/misc/bepasty.nix
index fad4827c6650..bda2de232088 100644
--- a/nixos/modules/services/misc/bepasty.nix
+++ b/nixos/modules/services/misc/bepasty.nix
@@ -68,6 +68,14 @@ in
               '';
           };
 
+          extraConfigFile = mkOption {
+            type = types.nullOr types.str;
+            description = ''
+              Path of file to be included as extra configuration.
+              '';
+            default = null;
+          };
+
           secretKey = mkOption {
             type = types.str;
             description = ''
@@ -154,6 +162,7 @@ in
               SECRET_KEY="$(cat "${server.secretKeyFile}")"
               DEFAULT_PERMISSIONS="${server.defaultPermissions}"
               ${server.extraConfig}
+              $(cat "${server.extraConfigFile}")
               EOF
             '';
             ExecStart = ''${gunicorn}/bin/gunicorn bepasty.wsgi --name ${name} \
