diff --git a/nixos/modules/config/terminfo.nix b/nixos/modules/config/terminfo.nix
index bf8bc43d6bb5..d5103de1066f 100644
--- a/nixos/modules/config/terminfo.nix
+++ b/nixos/modules/config/terminfo.nix
@@ -30,7 +30,6 @@
     #  pkgs)
     environment.systemPackages = lib.mkIf config.environment.enableAllTerminfo (map (x: x.terminfo) (with pkgs.pkgsBuildBuild; [
       alacritty
-      contour
       foot
       kitty
       mtm
