diff --git a/nixos/modules/services/misc/gitea.nix b/nixos/modules/services/misc/gitea.nix
index d4225232ed60..279fcc29b3dd 100644
--- a/nixos/modules/services/misc/gitea.nix
+++ b/nixos/modules/services/misc/gitea.nix
@@ -374,6 +374,20 @@ in
         description = "Path to a file containing the metrics authentication token.";
       };
 
+      storageMinioAccessKeyFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "/var/lib/secrets/gitea/storage_minio_access_key";
+        description = "Path to a file containing the Minio access key.";
+      };
+
+      storageMinioSecretKeyFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "/var/lib/secrets/gitea/storage_minio_secret_key";
+        description = "Path to a file containing the Minio secret key.";
+      };
+
       settings = mkOption {
         default = { };
         description = ''
@@ -653,10 +667,15 @@ in
           JWT_SECRET = "#oauth2jwtsecret#";
         };
 
-        lfs = mkIf cfg.lfs.enable {
+        lfs = mkIf (cfg.lfs.enable && !(cfg.storageMinioAccessKeyFile != null && cfg.storageMinioSecretKeyFile != null)) {
           PATH = cfg.lfs.contentDir;
         };
 
+        storage = mkIf (cfg.storageMinioAccessKeyFile != null && cfg.storageMinioSecretKeyFile != null) {
+          MINIO_ACCESS_KEY_ID = "#storageminioaccesskey#";
+          MINIO_SECRET_ACCESS_KEY = "#storageminiosecretkey#";
+        };
+
         packages.CHUNKED_UPLOAD_PATH = "${cfg.stateDir}/tmp/package-upload";
       };
 
@@ -801,6 +820,14 @@ in
                 ${replaceSecretBin} '#metricstoken#' '${cfg.metricsTokenFile}' '${runConfig}'
               ''}
 
+              ${lib.optionalString (cfg.storageMinioAccessKeyFile != null) ''
+                ${replaceSecretBin} '#storageminioaccesskey#' '${cfg.storageMinioAccessKeyFile}' '${runConfig}'
+              ''}
+
+              ${lib.optionalString (cfg.storageMinioSecretKeyFile != null) ''
+                ${replaceSecretBin} '#storageminiosecretkey#' '${cfg.storageMinioSecretKeyFile}' '${runConfig}'
+              ''}
+
               ${lib.optionalString (cfg.captcha.secretFile != null) ''
                 ${replaceSecretBin} '#captchasecret#' '${cfg.captcha.secretFile}' '${runConfig}'
               ''}
