diff --git a/nixos/modules/services/monitoring/grafana.nix b/nixos/modules/services/monitoring/grafana.nix
index 9b3068796d8..459812a1d2d 100644
--- a/nixos/modules/services/monitoring/grafana.nix
+++ b/nixos/modules/services/monitoring/grafana.nix
@@ -352,7 +352,7 @@ in {
               '';
               default = provisionConfDir;
               defaultText = "directory with links to files generated from services.grafana.provision";
-              type = types.path;
+              type = types.str;
             };
           };
 
