diff --git a/nixos/modules/services/monitoring/grafana.nix b/nixos/modules/services/monitoring/grafana.nix
index 9e15844f7dd..181f2efb206 100644
--- a/nixos/modules/services/monitoring/grafana.nix
+++ b/nixos/modules/services/monitoring/grafana.nix
@@ -343,7 +343,7 @@ in {
                   ln -sf ''${muteTimingsFile} $out/alerting/muteTimings.yaml
                   \'\'
               '';
-              type = types.path;
+              type = types.str;
             };
           };
 
