diff --git a/nixos/modules/services/x11/desktop-managers/plasma5.nix b/nixos/modules/services/x11/desktop-managers/plasma5.nix
index 65b94f61c4ae..e89c7340a7e4 100644
--- a/nixos/modules/services/x11/desktop-managers/plasma5.nix
+++ b/nixos/modules/services/x11/desktop-managers/plasma5.nix
@@ -388,7 +388,8 @@ in
         ++ lib.optional config.services.colord.enable pkgs.colord-kde
         ++ lib.optional config.services.hardware.bolt.enable pkgs.plasma5Packages.plasma-thunderbolt
         ++ lib.optionals config.services.samba.enable [ kdenetwork-filesharing pkgs.samba ]
-        ++ lib.optional config.services.xserver.wacom.enable pkgs.wacomtablet;
+        ++ lib.optional config.services.xserver.wacom.enable pkgs.wacomtablet
+        ++ lib.optional config.services.flatpak.enable flatpak-kcm;
 
       # Extra services for D-Bus activation
       services.dbus.packages = [
diff --git a/pkgs/desktops/plasma-5/default.nix b/pkgs/desktops/plasma-5/default.nix
index dcc5c7f7cf3d..a0be6b9ee8c1 100644
--- a/pkgs/desktops/plasma-5/default.nix
+++ b/pkgs/desktops/plasma-5/default.nix
@@ -123,6 +123,7 @@ let
       breeze-grub = callPackage ./breeze-grub.nix { };
       breeze-plymouth = callPackage ./breeze-plymouth { };
       discover = callPackage ./discover.nix { };
+      flatpak-kcm = callPackage ./flatpak-kcm.nix { };
       kactivitymanagerd = callPackage ./kactivitymanagerd.nix { };
       kde-cli-tools = callPackage ./kde-cli-tools.nix { };
       kde-gtk-config = callPackage ./kde-gtk-config { inherit gsettings-desktop-schemas; };
@@ -162,6 +163,7 @@ let
       plasma-systemmonitor = callPackage ./plasma-systemmonitor.nix { };
       plasma-thunderbolt = callPackage ./plasma-thunderbolt.nix { };
       plasma-vault = callPackage ./plasma-vault { };
+      plasma-welcome = callPackage ./plasma-welcome.nix { };
       plasma-workspace = callPackage ./plasma-workspace { };
       plasma-workspace-wallpapers = callPackage ./plasma-workspace-wallpapers.nix { };
       polkit-kde-agent = callPackage ./polkit-kde-agent.nix { };
diff --git a/pkgs/desktops/plasma-5/fetch.sh b/pkgs/desktops/plasma-5/fetch.sh
index 2da4dc40e6be..ba45a3e593a5 100644
--- a/pkgs/desktops/plasma-5/fetch.sh
+++ b/pkgs/desktops/plasma-5/fetch.sh
@@ -1 +1 @@
-WGET_ARGS=( https://download.kde.org/stable/plasma/5.26.5/ -A '*.tar.xz' )
+WGET_ARGS=( https://download.kde.org/unstable/plasma/5.26.90/ -A '*.tar.xz' )
diff --git a/pkgs/desktops/plasma-5/flatpak-kcm.nix b/pkgs/desktops/plasma-5/flatpak-kcm.nix
new file mode 100644
index 000000000000..406a0c9c920c
--- /dev/null
+++ b/pkgs/desktops/plasma-5/flatpak-kcm.nix
@@ -0,0 +1,18 @@
+{ mkDerivation
+, extra-cmake-modules
+, flatpak
+, kcmutils
+, kconfig
+, kdeclarative
+}:
+
+mkDerivation {
+  pname = "flatpak-kcm";
+  nativeBuildInputs = [ extra-cmake-modules ];
+  buildInputs = [
+    flatpak
+    kcmutils
+    kconfig
+    kdeclarative
+  ];
+}
diff --git a/pkgs/desktops/plasma-5/kscreenlocker.nix b/pkgs/desktops/plasma-5/kscreenlocker.nix
index d53d808e1084..f09104d534f9 100644
--- a/pkgs/desktops/plasma-5/kscreenlocker.nix
+++ b/pkgs/desktops/plasma-5/kscreenlocker.nix
@@ -7,6 +7,7 @@
 , kdeclarative
 , kglobalaccel
 , kidletime
+, libkscreen
 , kwayland
 , libXcursor
 , pam
@@ -27,6 +28,7 @@ mkDerivation {
     kdeclarative
     kglobalaccel
     kidletime
+    libkscreen
     kwayland
     libXcursor
     pam
diff --git a/pkgs/desktops/plasma-5/kwin/0003-plugins-qpa-allow-using-nixos-wrapper.patch b/pkgs/desktops/plasma-5/kwin/0003-plugins-qpa-allow-using-nixos-wrapper.patch
index 50c7cef3f8cd..d0be721b044c 100644
--- a/pkgs/desktops/plasma-5/kwin/0003-plugins-qpa-allow-using-nixos-wrapper.patch
+++ b/pkgs/desktops/plasma-5/kwin/0003-plugins-qpa-allow-using-nixos-wrapper.patch
@@ -13,9 +13,9 @@ index efd236b..a69c046 100644
 --- a/src/plugins/qpa/main.cpp
 +++ b/src/plugins/qpa/main.cpp
 @@ -23,7 +23,7 @@ public:
+
  QPlatformIntegration *KWinIntegrationPlugin::create(const QString &system, const QStringList &paramList)
  {
-     Q_UNUSED(paramList)
 -    if (!QCoreApplication::applicationFilePath().endsWith(QLatin1String("kwin_wayland")) && !qEnvironmentVariableIsSet("KWIN_FORCE_OWN_QPA")) {
 +    if (!QCoreApplication::applicationFilePath().endsWith(QLatin1String("kwin_wayland")) && !QCoreApplication::applicationFilePath().endsWith(QLatin1String(".kwin_wayland-wrapped")) && !qEnvironmentVariableIsSet("KWIN_FORCE_OWN_QPA")) {
          // Not KWin
diff --git a/pkgs/desktops/plasma-5/libkscreen/libkscreen-backends-path.patch b/pkgs/desktops/plasma-5/libkscreen/libkscreen-backends-path.patch
index 3d19f2d89fa1..948c045db4b1 100644
--- a/pkgs/desktops/plasma-5/libkscreen/libkscreen-backends-path.patch
+++ b/pkgs/desktops/plasma-5/libkscreen/libkscreen-backends-path.patch
@@ -1,16 +1,20 @@
-Index: libkscreen-5.9.4/src/backendmanager.cpp
-===================================================================
---- libkscreen-5.9.4.orig/src/backendmanager.cpp
-+++ libkscreen-5.9.4/src/backendmanager.cpp
-@@ -172,14 +172,11 @@ QFileInfo BackendManager::preferredBackend(const QString &backend)
+diff --git a/src/backendmanager.cpp b/src/backendmanager.cpp
+index e1013d5..4bded53 100644
+--- a/src/backendmanager.cpp
++++ b/src/backendmanager.cpp
+@@ -164,18 +164,11 @@ QFileInfo BackendManager::preferredBackend(const QString &backend)
+ 
  QFileInfoList BackendManager::listBackends()
  {
-     // Compile a list of installed backends first
+-    // Compile a list of installed backends first
 -    const QString backendFilter = QStringLiteral("KSC_*");
 -    const QStringList paths = QCoreApplication::libraryPaths();
 -    QFileInfoList finfos;
 -    for (const QString &path : paths) {
--        const QDir dir(path + QLatin1String("/kf5/kscreen/"), backendFilter, QDir::SortFlags(QDir::QDir::Name), QDir::NoDotAndDotDot | QDir::Files);
+-        const QDir dir(path + QStringLiteral("/kf" QT_STRINGIFY(QT_VERSION_MAJOR) "/kscreen/"),
+-                       backendFilter,
+-                       QDir::SortFlags(QDir::QDir::Name),
+-                       QDir::NoDotAndDotDot | QDir::Files);
 -        finfos.append(dir.entryInfoList());
 -    }
 -    return finfos;
@@ -21,4 +25,4 @@ Index: libkscreen-5.9.4/src/backendmanager.cpp
 +    return dir.entryInfoList();
  }
  
- KScreen::AbstractBackend *BackendManager::loadBackendPlugin(QPluginLoader *loader, const QString &name, const QVariantMap &arguments)
+ void BackendManager::setBackendArgs(const QVariantMap &arguments)
diff --git a/pkgs/desktops/plasma-5/plasma-remotecontrollers.nix b/pkgs/desktops/plasma-5/plasma-remotecontrollers.nix
index 3f169244fdaf..be1c6ff42eff 100644
--- a/pkgs/desktops/plasma-5/plasma-remotecontrollers.nix
+++ b/pkgs/desktops/plasma-5/plasma-remotecontrollers.nix
@@ -8,6 +8,7 @@
 , kdeclarative
 , kcmutils
 , kpackage
+, kscreenlocker
 , kwindowsystem
 , wayland
 , pkg-config
@@ -29,6 +30,7 @@ mkDerivation {
     kdeclarative
     kcmutils
     kpackage
+    kscreenlocker
     kwindowsystem
     wayland
     libcec
diff --git a/pkgs/desktops/plasma-5/plasma-welcome.nix b/pkgs/desktops/plasma-5/plasma-welcome.nix
new file mode 100644
index 000000000000..484447d871b5
--- /dev/null
+++ b/pkgs/desktops/plasma-5/plasma-welcome.nix
@@ -0,0 +1,42 @@
+{ mkDerivation
+, extra-cmake-modules
+, qtquickcontrols2
+, accounts-qt
+, kaccounts-integration
+, kcoreaddons
+, kconfigwidgets
+, kdbusaddons
+, kdeclarative
+, ki18n
+, kio
+, kirigami2
+, knotifications
+, kservice
+, kuserfeedback
+, kwindowsystem
+, plasma-framework
+, signond
+}:
+
+mkDerivation {
+  pname = "plasma-welcome";
+  nativeBuildInputs = [ extra-cmake-modules ];
+  buildInputs = [
+    qtquickcontrols2
+    accounts-qt
+    kaccounts-integration
+    kcoreaddons
+    kconfigwidgets
+    kdbusaddons
+    kdeclarative
+    ki18n
+    kio
+    kirigami2
+    knotifications
+    kservice
+    kuserfeedback
+    kwindowsystem
+    plasma-framework
+    signond
+  ];
+}
diff --git a/pkgs/desktops/plasma-5/srcs.nix b/pkgs/desktops/plasma-5/srcs.nix
index cc0a02b6aabe..42c71dd78c37 100644
--- a/pkgs/desktops/plasma-5/srcs.nix
+++ b/pkgs/desktops/plasma-5/srcs.nix
@@ -4,467 +4,483 @@
 
 {
   aura-browser = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/aura-browser-5.26.5.tar.xz";
-      sha256 = "0dhj058vh577jyhcjdp6x9dmh6apxk6yinknfci8l6vss9gchini";
-      name = "aura-browser-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/aura-browser-5.26.90.tar.xz";
+      sha256 = "1zyddfsmdvgrlpjsw1r6f0bd29mmmsjj2vh39vrhn51nqn62jzqg";
+      name = "aura-browser-5.26.90.tar.xz";
     };
   };
   bluedevil = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/bluedevil-5.26.5.tar.xz";
-      sha256 = "069hsn40zbnpkd73zcnnkdjv7n30dqyls5yvg3m0ljphn1syi79y";
-      name = "bluedevil-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/bluedevil-5.26.90.tar.xz";
+      sha256 = "1dapgwqxxq876pmyigh103nh36wkhry1w64svsb6zp88qyy2c7xb";
+      name = "bluedevil-5.26.90.tar.xz";
     };
   };
   breeze = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/breeze-5.26.5.tar.xz";
-      sha256 = "00zzb03jan14byh19mga3jrzwi3rl084cgpnk0wj3gwgf903p78y";
-      name = "breeze-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/breeze-5.26.90.tar.xz";
+      sha256 = "001p57rf7nabq3xc5ipv87c70cmq1cddayk8bscwk0y2wssaza7y";
+      name = "breeze-5.26.90.tar.xz";
     };
   };
   breeze-grub = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/breeze-grub-5.26.5.tar.xz";
-      sha256 = "0ibjqv6hyqi4ljf5jhx5sv55li9jcvk5lydl0xrb54v8czbarg4w";
-      name = "breeze-grub-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/breeze-grub-5.26.90.tar.xz";
+      sha256 = "1csig5l8np96apqdamwyxrzn2pwis4gyyki9vzxrsja5lssbrddh";
+      name = "breeze-grub-5.26.90.tar.xz";
     };
   };
   breeze-gtk = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/breeze-gtk-5.26.5.tar.xz";
-      sha256 = "1k1kcrv6cbx9m0yxnqhdgwq0bd8qf8055acwjdphwjhl50icliqi";
-      name = "breeze-gtk-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/breeze-gtk-5.26.90.tar.xz";
+      sha256 = "0kmxwqb83d2cf75afqg2f3dywklrryxnn3djwa18kmxnqmqymrrl";
+      name = "breeze-gtk-5.26.90.tar.xz";
     };
   };
   breeze-plymouth = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/breeze-plymouth-5.26.5.tar.xz";
-      sha256 = "0skiii7ni6r6jp3a7nq10hd6qdria1481lml4l57ma9afgvpxi9v";
-      name = "breeze-plymouth-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/breeze-plymouth-5.26.90.tar.xz";
+      sha256 = "0h35nplg0z30gq6704b759fs0prqrdrpabl1922xdidv8ysx3q19";
+      name = "breeze-plymouth-5.26.90.tar.xz";
     };
   };
   discover = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/discover-5.26.5.tar.xz";
-      sha256 = "19fd4f7nvqznfkrivxh77si627kqjnm1g17lidl9alz2iy2bxfm6";
-      name = "discover-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/discover-5.26.90.tar.xz";
+      sha256 = "0byi4sd5y1vfjs9331aw2rrb0j0ddh51p5041zhg5kg6ghgc3wkd";
+      name = "discover-5.26.90.tar.xz";
     };
   };
   drkonqi = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/drkonqi-5.26.5.tar.xz";
-      sha256 = "1syl78sl7wzjdb53glln6laa017cawgz0c39p33bdwq3kpzckw07";
-      name = "drkonqi-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/drkonqi-5.26.90.tar.xz";
+      sha256 = "0svf6as1c82h83z0gpanvsn1wxwhvx9sw2zzhymsyxrn6892v2sb";
+      name = "drkonqi-5.26.90.tar.xz";
+    };
+  };
+  flatpak-kcm = {
+    version = "5.26.90";
+    src = fetchurl {
+      url = "${mirror}/unstable/plasma/5.26.90/flatpak-kcm-5.26.90.tar.xz";
+      sha256 = "1dxps1bfa6v691awhb9z3jq0xj1zzpzn078pmc16msyp5k7ycqmz";
+      name = "flatpak-kcm-5.26.90.tar.xz";
     };
   };
   kactivitymanagerd = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kactivitymanagerd-5.26.5.tar.xz";
-      sha256 = "0ypnnh3lpcfw8ggi6nc8sr4hcsamx6s3ax869b6rhh392avcw2ha";
-      name = "kactivitymanagerd-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kactivitymanagerd-5.26.90.tar.xz";
+      sha256 = "1m98gzqikw94yr9221z460v32638kq4avphh60mz86znzasrcx10";
+      name = "kactivitymanagerd-5.26.90.tar.xz";
     };
   };
   kde-cli-tools = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kde-cli-tools-5.26.5.tar.xz";
-      sha256 = "0djxqxdkih4svmggk6lk6p6n3sx3w4q9fsnx2dxfn5injwh0qrkr";
-      name = "kde-cli-tools-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kde-cli-tools-5.26.90.tar.xz";
+      sha256 = "0l77h57jvh4fpn6klmlxjk7sqhrazp8rmji8awwb9kzmm0j9fxrl";
+      name = "kde-cli-tools-5.26.90.tar.xz";
     };
   };
   kde-gtk-config = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kde-gtk-config-5.26.5.tar.xz";
-      sha256 = "10r4kgpd04hh05yriga2r5awwdb8v93jsx0qksdgj3661xy00x7n";
-      name = "kde-gtk-config-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kde-gtk-config-5.26.90.tar.xz";
+      sha256 = "03ya8jgdjfpsl2h3xxl49vi710ss57hjfpmibj4sdzvfjf9qvcgb";
+      name = "kde-gtk-config-5.26.90.tar.xz";
     };
   };
   kdecoration = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kdecoration-5.26.5.tar.xz";
-      sha256 = "1dzpqzi25wxmilj92lqjmd6wid3nlkbvnnpcpp0ywd860q30zzd2";
-      name = "kdecoration-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kdecoration-5.26.90.tar.xz";
+      sha256 = "15x33fhh8d1y67pqg7dx47wzk2j4fmwmq459j1bv80s511zsv34v";
+      name = "kdecoration-5.26.90.tar.xz";
     };
   };
   kdeplasma-addons = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kdeplasma-addons-5.26.5.tar.xz";
-      sha256 = "17vrdnigif4v38gbh04p8qs7wqp3y3nqll1m7xc4qfqbq1dydyjq";
-      name = "kdeplasma-addons-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kdeplasma-addons-5.26.90.tar.xz";
+      sha256 = "0l1f7cbpgciskr0i7l77nkdff6cnfyh9gy69z4jds6isswv23b07";
+      name = "kdeplasma-addons-5.26.90.tar.xz";
     };
   };
   kgamma5 = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kgamma5-5.26.5.tar.xz";
-      sha256 = "0v4rpm2asv31w1rqrhn25kbcdl1acahg10lmh6iwisxl1i3x8hd8";
-      name = "kgamma5-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kgamma5-5.26.90.tar.xz";
+      sha256 = "0bf00dp70lvnaabldhwspj5ml3jjvldklr7rsgxscvgy2v228gw3";
+      name = "kgamma5-5.26.90.tar.xz";
     };
   };
   khotkeys = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/khotkeys-5.26.5.tar.xz";
-      sha256 = "0lkmh5n76kmrdxm4snkk1zz7d927qkl69ajd5qhz277whpz3slw8";
-      name = "khotkeys-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/khotkeys-5.26.90.tar.xz";
+      sha256 = "12gg30hxc0559q4185lndqgwf98ayjqgsk7f6h32yqh0q5mlg6hx";
+      name = "khotkeys-5.26.90.tar.xz";
     };
   };
   kinfocenter = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kinfocenter-5.26.5.tar.xz";
-      sha256 = "0rka2fm7r2zikv3lcd1q39swgcp6fs5l4f5ldpgwpk9bps96cy4x";
-      name = "kinfocenter-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kinfocenter-5.26.90.tar.xz";
+      sha256 = "1hmm0ji0d4gbx9ldsm8n8cb2xkda54m44zf3sca579pc3h3g4nja";
+      name = "kinfocenter-5.26.90.tar.xz";
     };
   };
   kmenuedit = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kmenuedit-5.26.5.tar.xz";
-      sha256 = "17iyk77mb1lp9xjl92ffvv111b4m2qq98xznd1qafnah7ix1cg7p";
-      name = "kmenuedit-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kmenuedit-5.26.90.tar.xz";
+      sha256 = "1zaji53s4zq6zz2syz96w900a8gx69i6pflcqxrkdldcbccky6h2";
+      name = "kmenuedit-5.26.90.tar.xz";
     };
   };
   kpipewire = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kpipewire-5.26.5.tar.xz";
-      sha256 = "1yqymwq48mqv9zfw3idb0abizrfhr7f4d37ys9w7a9jixcm7rdrv";
-      name = "kpipewire-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kpipewire-5.26.90.tar.xz";
+      sha256 = "15cr3jbk2nlchglk9s06872hhw2wr75kiyjyh7y430y3x9hzdqd6";
+      name = "kpipewire-5.26.90.tar.xz";
     };
   };
   kscreen = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kscreen-5.26.5.tar.xz";
-      sha256 = "0w8yrzpgr23s7sr3296lrm2zbm2j2174zv5az41i44l907iq9y1x";
-      name = "kscreen-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kscreen-5.26.90.tar.xz";
+      sha256 = "046qbkq9wmfl1arigb69wg8fv566h6jn9z2ifchwmn3gg6dplwl4";
+      name = "kscreen-5.26.90.tar.xz";
     };
   };
   kscreenlocker = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kscreenlocker-5.26.5.tar.xz";
-      sha256 = "0lwn3xbbk1578974m3nkk2x5h9fadrcizvh4q4528s5wgxja6w2g";
-      name = "kscreenlocker-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kscreenlocker-5.26.90.tar.xz";
+      sha256 = "09chf3wnb02zn0iqn0cjbnyqvb90ziry026iivd00zxl97q8m0kj";
+      name = "kscreenlocker-5.26.90.tar.xz";
     };
   };
   ksshaskpass = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/ksshaskpass-5.26.5.tar.xz";
-      sha256 = "0k0jxbkav115i5nhazxyz3s3wg6gyfcsa1jilf3j3fr8vnh8qhx0";
-      name = "ksshaskpass-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/ksshaskpass-5.26.90.tar.xz";
+      sha256 = "01q8a3llg8jzdvwhpbziqwllk12nb7paqa7h12yai5s81vagqxja";
+      name = "ksshaskpass-5.26.90.tar.xz";
     };
   };
   ksystemstats = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/ksystemstats-5.26.5.tar.xz";
-      sha256 = "18sa2w6vgbqmw8wrg5vqbraql04x9kb7mj9k35z4zs2n4abfrrbi";
-      name = "ksystemstats-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/ksystemstats-5.26.90.tar.xz";
+      sha256 = "1mz5cmzx1y2m6r3a924xvfyfmqjn1fqx1nvkvm3qjzkyl08db366";
+      name = "ksystemstats-5.26.90.tar.xz";
     };
   };
   kwallet-pam = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kwallet-pam-5.26.5.tar.xz";
-      sha256 = "1c5pzih8id0gld7qdf8ffxsbfgq9bsvpx37qyzr941b3vif7f9yx";
-      name = "kwallet-pam-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kwallet-pam-5.26.90.tar.xz";
+      sha256 = "1wmgwx1gk4q24rb7hxn5mxkk5iys09s1gkadb0j60yyzfl6fz68r";
+      name = "kwallet-pam-5.26.90.tar.xz";
     };
   };
   kwayland-integration = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kwayland-integration-5.26.5.tar.xz";
-      sha256 = "0czq16lq210hq7a3wqg7pvdr9a8vm16z6yb82p0w6yqv15kk7alx";
-      name = "kwayland-integration-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kwayland-integration-5.26.90.tar.xz";
+      sha256 = "1mik6y34s0dg6579wrjcsjwqvbip2hrxk3amwgr3315xwprvjs0k";
+      name = "kwayland-integration-5.26.90.tar.xz";
     };
   };
   kwin = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kwin-5.26.5.tar.xz";
-      sha256 = "1n0v0553s9l8s6wgdz5bhrnvww20q49h92v3m4w216razj7q3q0k";
-      name = "kwin-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kwin-5.26.90.tar.xz";
+      sha256 = "1460kr2dmmqbalklyk1kdj7fsw8rk0k32y5plhvrgxkvd0m1nvia";
+      name = "kwin-5.26.90.tar.xz";
     };
   };
   kwrited = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/kwrited-5.26.5.tar.xz";
-      sha256 = "14savlbywkx5107s0viprafs9pl389zhjk32iy4x5jmf9dk6vmm9";
-      name = "kwrited-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/kwrited-5.26.90.tar.xz";
+      sha256 = "1kvk07wvajl3vw5qqgy4zn4f0in9dl7ywdzhj4cpqnz7rdhxrrr9";
+      name = "kwrited-5.26.90.tar.xz";
     };
   };
   layer-shell-qt = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/layer-shell-qt-5.26.5.tar.xz";
-      sha256 = "1k79dcyic2hkm92sd7aix03ywagr4rgsdf86a9zy06kafg92jra9";
-      name = "layer-shell-qt-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/layer-shell-qt-5.26.90.tar.xz";
+      sha256 = "1zmchljclcgdx0m4dln2485wia6bmr00xfwpyq8553waxp7p0iy4";
+      name = "layer-shell-qt-5.26.90.tar.xz";
     };
   };
   libkscreen = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/libkscreen-5.26.5.tar.xz";
-      sha256 = "0vajgzjvpn909ds88jddsg7bman0rdphrm70b0k5ar7bcw0x54d5";
-      name = "libkscreen-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/libkscreen-5.26.90.tar.xz";
+      sha256 = "0lmrwgsl71agxfxpyyqp7viljcxrcqmxxaq1i1vs3l2znqdmw021";
+      name = "libkscreen-5.26.90.tar.xz";
     };
   };
   libksysguard = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/libksysguard-5.26.5.tar.xz";
-      sha256 = "01a1wm6z191lzjqd5iws90jd3ld3r6r1qaw95bl65l9ar8p3gr8p";
-      name = "libksysguard-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/libksysguard-5.26.90.tar.xz";
+      sha256 = "0a867fp15m72laa705lhb0v4ycbkdv5xvc44xklcbx8rbxmpdrjl";
+      name = "libksysguard-5.26.90.tar.xz";
     };
   };
   milou = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/milou-5.26.5.tar.xz";
-      sha256 = "1pmz4n41ysb0mhgykvc65r5as8fkd7wr4k57sk0a12wzsndy7zv8";
-      name = "milou-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/milou-5.26.90.tar.xz";
+      sha256 = "1h3jh5xkp34ibznwnmwbqaqpd3ln7fvdrc2xmpbnvqv03y7y761p";
+      name = "milou-5.26.90.tar.xz";
     };
   };
   oxygen = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/oxygen-5.26.5.tar.xz";
-      sha256 = "0wllhniaqpzvzji51a9871j1697kjwbz4xj8q9x1w89ip88pb6gq";
-      name = "oxygen-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/oxygen-5.26.90.tar.xz";
+      sha256 = "1222kgyd24z335d1qgqwf8i5mm4j00cpamyl9kbdkhi8dq1chmj8";
+      name = "oxygen-5.26.90.tar.xz";
     };
   };
   oxygen-sounds = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/oxygen-sounds-5.26.5.tar.xz";
-      sha256 = "1ws6bssn2r2m666kd6i0xnmzxb8x3z2pa6qk36xpbs1xjxdv5wny";
-      name = "oxygen-sounds-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/oxygen-sounds-5.26.90.tar.xz";
+      sha256 = "07dbm7ah3bj3brqcd4m2xa2x6a28bpdz33dbk9lscy7dr930bzj7";
+      name = "oxygen-sounds-5.26.90.tar.xz";
     };
   };
   plank-player = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plank-player-5.26.5.tar.xz";
-      sha256 = "18yw74mqcd0igps10m1sd41rlir2y8y78xkljxj5w2y4yvh97vbs";
-      name = "plank-player-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plank-player-5.26.90.tar.xz";
+      sha256 = "15p1lb14n5k40ngq8x6wqqxq3mg0b78y46nv0h26v2vv6ny1x6yq";
+      name = "plank-player-5.26.90.tar.xz";
     };
   };
   plasma-bigscreen = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-bigscreen-5.26.5.tar.xz";
-      sha256 = "0893ngwdgq1l76gib4p9212lwp4i2gxvpnmxygndbmkz7nhjax75";
-      name = "plasma-bigscreen-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-bigscreen-5.26.90.tar.xz";
+      sha256 = "1n9sd0d1s31l29c9jy6gavp8dsrvsi3im4978dw56xybcc8mgm2c";
+      name = "plasma-bigscreen-5.26.90.tar.xz";
     };
   };
   plasma-browser-integration = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-browser-integration-5.26.5.tar.xz";
-      sha256 = "1bfzha5vd8hjij758b0g6ibyi4f8gmijz92fgn5is0p1hydm8y4l";
-      name = "plasma-browser-integration-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-browser-integration-5.26.90.tar.xz";
+      sha256 = "04fal5x7sh6ar8xfx5nsdl1fwqinvf3a572cbxbfqkiwf400nqmg";
+      name = "plasma-browser-integration-5.26.90.tar.xz";
     };
   };
   plasma-desktop = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-desktop-5.26.5.tar.xz";
-      sha256 = "15kc5sam3y2c1ccp14cs7v5zakgzz9gpxpyxapimlvkwscxmvzn6";
-      name = "plasma-desktop-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-desktop-5.26.90.tar.xz";
+      sha256 = "05wycirq8c2w2b3j99nvr7cwl2ki2ihzkp87wbs4h9xg95zsgvq8";
+      name = "plasma-desktop-5.26.90.tar.xz";
     };
   };
   plasma-disks = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-disks-5.26.5.tar.xz";
-      sha256 = "137k44lly2qv6vrmfi4g2xqc7y907ywwshp7mn07qmbps3a85h30";
-      name = "plasma-disks-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-disks-5.26.90.tar.xz";
+      sha256 = "14f1zlz8bfaazrjrslri2cyhhw2xkqv30d9v1k4m3h79x9b60qvg";
+      name = "plasma-disks-5.26.90.tar.xz";
     };
   };
   plasma-firewall = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-firewall-5.26.5.tar.xz";
-      sha256 = "18fp7iqnyjx2p44n8hjs7fss212ja3k2i5cryrnmwpf7r0q73l36";
-      name = "plasma-firewall-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-firewall-5.26.90.tar.xz";
+      sha256 = "05zmc0yx67h4bbvy6fzxkrymrnii9k54q3pczjp4wsbir6mxni5y";
+      name = "plasma-firewall-5.26.90.tar.xz";
     };
   };
   plasma-integration = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-integration-5.26.5.tar.xz";
-      sha256 = "0ap7rzhqmm7ai9sc6w281f2kf81qgs2h2qpf5bvbc6wm3vn1lm1g";
-      name = "plasma-integration-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-integration-5.26.90.tar.xz";
+      sha256 = "11fsx5hkjlr83ad9idn9rrgc8y0saj1c6pk2z2whgirq9bw9yab7";
+      name = "plasma-integration-5.26.90.tar.xz";
     };
   };
   plasma-mobile = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-mobile-5.26.5.tar.xz";
-      sha256 = "06gha8mga9xqmmvzc9why9pfzzpy8czrqym3k3n23pa9hy2qqsvj";
-      name = "plasma-mobile-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-mobile-5.26.90.tar.xz";
+      sha256 = "16wpkwbxb1jkicm9m50mwaphyk0cg3z5svd84lc2izxqclwwqwy7";
+      name = "plasma-mobile-5.26.90.tar.xz";
     };
   };
   plasma-nano = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-nano-5.26.5.tar.xz";
-      sha256 = "161sabckgl8gjxxsm0aklp33bf05iq2p84vh2v5y6bfpimsnamnj";
-      name = "plasma-nano-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-nano-5.26.90.tar.xz";
+      sha256 = "06ipr8jdjfrkcx97w095g4isrbrs69mbs2a7d78r7a63vgmd10bm";
+      name = "plasma-nano-5.26.90.tar.xz";
     };
   };
   plasma-nm = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-nm-5.26.5.tar.xz";
-      sha256 = "1msi9lhkhpb29h6fwkxqmp9js70r8vrvzzf38hzkgxcbf3757k64";
-      name = "plasma-nm-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-nm-5.26.90.tar.xz";
+      sha256 = "1ckyaf5fddjc04mlv57an5w0rcxdygv2nfyywmadd5zbgnahyc6c";
+      name = "plasma-nm-5.26.90.tar.xz";
     };
   };
   plasma-pa = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-pa-5.26.5.tar.xz";
-      sha256 = "19i9ipajiz68silvprap9pykki6rhb7d2ways501bx8g4vfcb0h5";
-      name = "plasma-pa-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-pa-5.26.90.tar.xz";
+      sha256 = "1gw7xisn3ix52wmh314wy8xnd9rs171rimglqdnvlgjqrymil9ky";
+      name = "plasma-pa-5.26.90.tar.xz";
     };
   };
   plasma-remotecontrollers = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-remotecontrollers-5.26.5.tar.xz";
-      sha256 = "1cam1jchp3nwqkqskln4qbpb4rixzwa0bwma89ysdgqy0pzpq2x0";
-      name = "plasma-remotecontrollers-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-remotecontrollers-5.26.90.tar.xz";
+      sha256 = "0wysl3r8xq8q1qf5ja0dwx0b4y7saqd0433hvn06cp8y579190xh";
+      name = "plasma-remotecontrollers-5.26.90.tar.xz";
     };
   };
   plasma-sdk = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-sdk-5.26.5.tar.xz";
-      sha256 = "0p20s2cp5vd1chj28wq88ixb625hisl134hnlllz6h70s5vzy79n";
-      name = "plasma-sdk-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-sdk-5.26.90.tar.xz";
+      sha256 = "0ykxgx5j5s28byshr65xp62wbpfp9saya18chymwak5jk2ylndkw";
+      name = "plasma-sdk-5.26.90.tar.xz";
     };
   };
   plasma-systemmonitor = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-systemmonitor-5.26.5.tar.xz";
-      sha256 = "0xghszlm8h70p192qizv2kj7f02fnf3p3hg1p10ni7sqfaczrbq1";
-      name = "plasma-systemmonitor-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-systemmonitor-5.26.90.tar.xz";
+      sha256 = "1ms40gjygq360wak3v412by0kvw9zy58v4p85phghzakc8r2r5gc";
+      name = "plasma-systemmonitor-5.26.90.tar.xz";
     };
   };
   plasma-tests = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-tests-5.26.5.tar.xz";
-      sha256 = "1h0cinrgjdp7rhl3v55c40a5agd93rqwx1xjqyj06vkjkbigx0fr";
-      name = "plasma-tests-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-tests-5.26.90.tar.xz";
+      sha256 = "16hr2z5hcqkv5mv2gqjkrxcb5w89wzwxygj9sa7k4fmd94ga63vj";
+      name = "plasma-tests-5.26.90.tar.xz";
     };
   };
   plasma-thunderbolt = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-thunderbolt-5.26.5.tar.xz";
-      sha256 = "1lzj3m2k34m3fvcq3ykr5igaq0sbc4v87cncxvz6ln0p7cq0i6y5";
-      name = "plasma-thunderbolt-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-thunderbolt-5.26.90.tar.xz";
+      sha256 = "0qhbpjsms40a3q6qwl3h1n183n3789l9q5k6qlnlgs890m1kgzza";
+      name = "plasma-thunderbolt-5.26.90.tar.xz";
     };
   };
   plasma-vault = {
-    version = "5.26.5";
+    version = "5.26.90";
+    src = fetchurl {
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-vault-5.26.90.tar.xz";
+      sha256 = "1dchn4xdnn2021yjw2dqrrh4rwh79jsaybrb9c1b3xilysvk3cxb";
+      name = "plasma-vault-5.26.90.tar.xz";
+    };
+  };
+  plasma-welcome = {
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-vault-5.26.5.tar.xz";
-      sha256 = "104cs6prb8v5mccshz5brknfllyfbfnjzzpdrgk1xpzh5apwmhja";
-      name = "plasma-vault-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-welcome-5.26.90.tar.xz";
+      sha256 = "0ra58xm2af7l4hjdnwpwpd8p2vr4ank6gm8hipp8iq7d43wixr1j";
+      name = "plasma-welcome-5.26.90.tar.xz";
     };
   };
   plasma-workspace = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-workspace-5.26.5.tar.xz";
-      sha256 = "082jnlqgbp48bmyp1nbzg2d0p0pxkvra6xjv40kvh6k6rmpr5cjc";
-      name = "plasma-workspace-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-workspace-5.26.90.tar.xz";
+      sha256 = "0zqsvqdw2fa9jxidqmhflfv6f6s71vpc563c3bnjqjd21z86px3j";
+      name = "plasma-workspace-5.26.90.tar.xz";
     };
   };
   plasma-workspace-wallpapers = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plasma-workspace-wallpapers-5.26.5.tar.xz";
-      sha256 = "07gln5jl947qfpvrgiqa1fnw8lnnf4jg734ayybi927j059s3g8i";
-      name = "plasma-workspace-wallpapers-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plasma-workspace-wallpapers-5.26.90.tar.xz";
+      sha256 = "19dnd32ysi8ypm3fncpal75mapaxylmyb9kzcch6kydx5jzjl60q";
+      name = "plasma-workspace-wallpapers-5.26.90.tar.xz";
     };
   };
   plymouth-kcm = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/plymouth-kcm-5.26.5.tar.xz";
-      sha256 = "1isxs6gxbab2ddl22c3992gmdgy3mi5yjb9inv32ajzs6r7snrj1";
-      name = "plymouth-kcm-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/plymouth-kcm-5.26.90.tar.xz";
+      sha256 = "1f7z35pm783ynrzxj6bhysvgwlf20wwvl01vq1ksia6kyam3zzvx";
+      name = "plymouth-kcm-5.26.90.tar.xz";
     };
   };
   polkit-kde-agent = {
-    version = "1-5.26.5";
+    version = "1-5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/polkit-kde-agent-1-5.26.5.tar.xz";
-      sha256 = "19jrjx40m8g6hwq5k7ibbm0fmb803hxync2022gqbwzfrdb4493i";
-      name = "polkit-kde-agent-1-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/polkit-kde-agent-1-5.26.90.tar.xz";
+      sha256 = "0myvxfkg28687hrkz2823qi8sq6cq8dphn0qmw9l5bbip9lbz4k8";
+      name = "polkit-kde-agent-1-5.26.90.tar.xz";
     };
   };
   powerdevil = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/powerdevil-5.26.5.tar.xz";
-      sha256 = "063bqwdnpskcwqzpgcmjwgs9vpkn1gnq2nnzj2sdzrlflfrdsfvp";
-      name = "powerdevil-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/powerdevil-5.26.90.tar.xz";
+      sha256 = "1gmlxf3cznxpa2jkhh64w0lbzzacriywh5xx72s32f9s9c04v2z6";
+      name = "powerdevil-5.26.90.tar.xz";
     };
   };
   qqc2-breeze-style = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/qqc2-breeze-style-5.26.5.tar.xz";
-      sha256 = "1lhv4jlg15w8a88czq0fazaizjwva3yv8wv3scdnhhyfnbkch9mr";
-      name = "qqc2-breeze-style-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/qqc2-breeze-style-5.26.90.tar.xz";
+      sha256 = "1src27zw83l04w7awailz5f08gpw37rqz1fmfw8b5jzjqr38cs02";
+      name = "qqc2-breeze-style-5.26.90.tar.xz";
     };
   };
   sddm-kcm = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/sddm-kcm-5.26.5.tar.xz";
-      sha256 = "0cglb7dvwrkdpl5bgg91yfxrvqfpc2szj3qipcmd6ladsvbz8cxv";
-      name = "sddm-kcm-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/sddm-kcm-5.26.90.tar.xz";
+      sha256 = "0725az6iwjn1ccihd3bqx6mv3wg8cjlfpvkd7h5yx5i9www01l7m";
+      name = "sddm-kcm-5.26.90.tar.xz";
     };
   };
   systemsettings = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/systemsettings-5.26.5.tar.xz";
-      sha256 = "1jb1d6933gq07vgg3kzqna91cpcv87p38wrk70g17iz7pszjb3ns";
-      name = "systemsettings-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/systemsettings-5.26.90.tar.xz";
+      sha256 = "1z6gsb585w2z7yx194cbs9yxxmf12zj5f82hay0ckch61m3i2mq5";
+      name = "systemsettings-5.26.90.tar.xz";
     };
   };
   xdg-desktop-portal-kde = {
-    version = "5.26.5";
+    version = "5.26.90";
     src = fetchurl {
-      url = "${mirror}/stable/plasma/5.26.5/xdg-desktop-portal-kde-5.26.5.tar.xz";
-      sha256 = "1hznyl90d3xswnjs4qc6r5azw1z3pzkr7z507v696xhv41j8bzsj";
-      name = "xdg-desktop-portal-kde-5.26.5.tar.xz";
+      url = "${mirror}/unstable/plasma/5.26.90/xdg-desktop-portal-kde-5.26.90.tar.xz";
+      sha256 = "05glxcnyjr09bxb2xpw93r0hx2786vmprjwz39xmpdvm1wvghy4p";
+      name = "xdg-desktop-portal-kde-5.26.90.tar.xz";
     };
   };
 }
diff --git a/pkgs/development/libraries/kuserfeedback/default.nix b/pkgs/development/libraries/kuserfeedback/default.nix
new file mode 100644
index 000000000000..bea9f86fc90d
--- /dev/null
+++ b/pkgs/development/libraries/kuserfeedback/default.nix
@@ -0,0 +1,23 @@
+{ lib
+, mkDerivation
+, fetchurl
+, extra-cmake-modules
+, qttools
+}:
+mkDerivation rec {
+  pname = "kuserfeedback";
+  version = "1.2.0";
+
+  src = fetchurl {
+    url = "mirror://kde/stable/${pname}/${pname}-${version}.tar.xz";
+    hash = "sha256-dqrJIrFTJJsnRoCm9McsI47xTj3wS60Ay2QVixBj8mQ=";
+  };
+
+  nativeBuildInputs = [ extra-cmake-modules qttools ];
+
+  meta = with lib; {
+    license = [ licenses.mit ];
+    maintainers = [ maintainers.k900 ];
+    description = "Framework for collecting user feedback for apps via telemetry and surveys";
+  };
+}
diff --git a/pkgs/top-level/qt5-packages.nix b/pkgs/top-level/qt5-packages.nix
index f8139b7dab52..6366b2f2be32 100644
--- a/pkgs/top-level/qt5-packages.nix
+++ b/pkgs/top-level/qt5-packages.nix
@@ -116,6 +116,8 @@ in (kdeFrameworks // plasmaMobileGear // plasma5 // plasma5.thirdParty // kdeGea
 
   kquickimageedit = callPackage ../development/libraries/kquickimageedit { };
 
+  kuserfeedback = callPackage ../development/libraries/kuserfeedback { };
+
   kweathercore = libsForQt5.callPackage ../development/libraries/kweathercore { };
 
   ldutils = callPackage ../development/libraries/ldutils { };
