diff --git a/nixos/modules/services/networking/networkmanager.nix b/nixos/modules/services/networking/networkmanager.nix
index 06052496d6e4..5ff04a285d61 100644
--- a/nixos/modules/services/networking/networkmanager.nix
+++ b/nixos/modules/services/networking/networkmanager.nix
@@ -605,11 +605,6 @@ in
     systemd.packages = packages;
 
     systemd.tmpfiles.settings.networkmanager = {
-      ${configAttrs.keyfile.path}.d = {
-        mode = "0700";
-        user = "root";
-        group = "root";
-      };
       # for dnsmasq.leases
       "/var/lib/misc".d = {
         mode = "0755";
