diff --git a/nixos/modules/services/networking/networkmanager.nix b/nixos/modules/services/networking/networkmanager.nix
index 1f3773c70b00..b9c8b6b815fd 100644
--- a/nixos/modules/services/networking/networkmanager.nix
+++ b/nixos/modules/services/networking/networkmanager.nix
@@ -603,7 +603,6 @@ in
     systemd.packages = packages;
 
     systemd.tmpfiles.rules = [
-      "d /etc/NetworkManager/system-connections 0700 root root -"
       "d /var/lib/misc 0755 root root -" # for dnsmasq.leases
       # ppp isn't able to mkdir that directory at runtime
       "d /run/pppd/lock 0700 root root -"
