diff --git a/pkgs/kde/plasma/plasma-workspace/default.nix b/pkgs/kde/plasma/plasma-workspace/default.nix
index 964525c99c9f..818e4ec7defe 100644
--- a/pkgs/kde/plasma/plasma-workspace/default.nix
+++ b/pkgs/kde/plasma/plasma-workspace/default.nix
@@ -93,6 +93,9 @@ mkKdeDerivation {
 
     moveToOutput share/xsessions $sessions
     moveToOutput share/wayland-sessions $sessions
+
+    # Force remove all login splash screens
+    rm -rf $out/share/plasma/look-and-feel/*/contents/splash
   '';
 
   passthru.providedSessions = [
