diff --git a/nixos/modules/services/networking/resilio.nix b/nixos/modules/services/networking/resilio.nix
index cc9495bf238..d4175c27b24 100644
--- a/nixos/modules/services/networking/resilio.nix
+++ b/nixos/modules/services/networking/resilio.nix
@@ -54,12 +54,6 @@ let
 
   createConfig = pkgs.writeShellScriptBin "create-resilio-config" ''
     ${pkgs.jq}/bin/jq \
-      '.shared_folders |= map(.secret = $ARGS.named[.dir])' \
-      ${
-        lib.concatMapStringsSep " \\\n  "
-        (entry: ''--arg '${entry.dir}' "$(cat '${entry.secretFile}')"'')
-        sharedFoldersSecretFiles
-      } \
       <${configFile} \
       >${runConfigPath}
   '';
@@ -259,16 +253,6 @@ in
         }
       ];
 
-    users.users.rslsync = {
-      description     = "Resilio Sync Service user";
-      home            = cfg.storagePath;
-      createHome      = true;
-      uid             = config.ids.uids.rslsync;
-      group           = "rslsync";
-    };
-
-    users.groups.rslsync = {};
-
     systemd.services.resilio = with pkgs; {
       description = "Resilio Sync Service";
       wantedBy    = [ "multi-user.target" ];
@@ -279,9 +263,14 @@ in
         User      = "rslsync";
         RuntimeDirectory = "rslsync";
         ExecStartPre = "${createConfig}/bin/create-resilio-config";
-        ExecStart = ''
+        ExecStart = pkgs.writeShellScript "rslsync-start" ''
+          exec ${pkgs.ip2unix}/bin/ip2unix -r in,tcp,port=${toString cfg.httpListenPort},path=/run/rslsync/rslsync.sock \
           ${resilioSync}/bin/rslsync --nodaemon --config ${runConfigPath}
         '';
+        ExecStartPost = pkgs.writeShellScript "rslsync-post" ''
+          while [ ! -S /run/rslsync/rslsync.sock ]; do sleep 1; done
+          chmod 777 /run/rslsync/rslsync.sock
+        '';
       };
     };
   };
