diff --git a/nixos/modules/services/networking/resilio.nix b/nixos/modules/services/networking/resilio.nix
index 7f6358d00d0..dde6d50bbdd 100644
--- a/nixos/modules/services/networking/resilio.nix
+++ b/nixos/modules/services/networking/resilio.nix
@@ -264,16 +264,6 @@ in
         }
       ];
 
-    users.users.rslsync = {
-      description     = "Resilio Sync Service user";
-      home            = cfg.storagePath;
-      createHome      = true;
-      uid             = config.ids.uids.rslsync;
-      group           = "rslsync";
-    };
-
-    users.groups.rslsync = {};
-
     systemd.services.resilio = with pkgs; {
       description = "Resilio Sync Service";
       wantedBy    = [ "multi-user.target" ];
@@ -284,9 +274,14 @@ in
         User      = "rslsync";
         RuntimeDirectory = "rslsync";
         ExecStartPre = "${createConfig}/bin/create-resilio-config";
-        ExecStart = ''
+        ExecStart = pkgs.writeShellScript "rslsync-start" ''
+          exec ${pkgs.ip2unix}/bin/ip2unix -r in,tcp,port=${toString cfg.httpListenPort},path=/run/rslsync/rslsync.sock \
           ${resilioSync}/bin/rslsync --nodaemon --config ${runConfigPath}
         '';
+        ExecStartPost = pkgs.writeShellScript "rslsync-post" ''
+          while [ ! -S /run/rslsync/rslsync.sock ]; do sleep 1; done
+          chmod 777 /run/rslsync/rslsync.sock
+        '';
       };
     };
   };
