diff --git a/nixos/modules/services/networking/resilio.nix b/nixos/modules/services/networking/resilio.nix
index 7f6358d00d0..a0f70a3bd69 100644
--- a/nixos/modules/services/networking/resilio.nix
+++ b/nixos/modules/services/networking/resilio.nix
@@ -284,9 +284,14 @@ in
         User      = "rslsync";
         RuntimeDirectory = "rslsync";
         ExecStartPre = "${createConfig}/bin/create-resilio-config";
-        ExecStart = ''
+        ExecStart = pkgs.writeShellScript "rslsync-start" ''
+          exec ${pkgs.ip2unix}/bin/ip2unix -r in,tcp,port=${toString cfg.httpListenPort},path=/run/rslsync/rslsync.sock \
           ${resilioSync}/bin/rslsync --nodaemon --config ${runConfigPath}
         '';
+        ExecStartPost = pkgs.writeShellScript "rslsync-post" ''
+          while [ ! -S /run/rslsync/rslsync.sock ]; do sleep 1; done
+          chmod 777 /run/rslsync/rslsync.sock
+        '';
       };
     };
   };
