diff --git a/pkgs/development/python-modules/sentence-transformers/default.nix b/pkgs/development/python-modules/sentence-transformers/default.nix
index 1c4f1d5a2430..e82ecae7d164 100644
--- a/pkgs/development/python-modules/sentence-transformers/default.nix
+++ b/pkgs/development/python-modules/sentence-transformers/default.nix
@@ -123,9 +123,6 @@ buildPythonPackage rec {
     "tests/test_pretrained_stsb.py"
     "tests/test_sentence_transformer.py"
     "tests/test_train_stsb.py"
-  ]
-  ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    # Segfault
     "tests/test_sparse_tensor.py"
   ];
 
