--- a/nixos/modules/services/networking/suricata/default.nix
+++ b/nixos/modules/services/networking/suricata/default.nix
@@ -152,6 +152,16 @@
         List of rules that should be disabled.
       '';
     };
+
+    enabledRules = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      description = ''
+        List of rules that should be enabled (re-enabling rules disabled
+        via {option}`services.suricata.disabledRules` or disabled by
+        default in the ruleset).
+      '';
+    };
   };
 
   config =
@@ -230,7 +240,8 @@
               ${concatStringsSep "\n" enabledSourcesCmds}
               ${python.interpreter} ${pkg}/bin/suricata-update update-sources
               ${python.interpreter} ${pkg}/bin/suricata-update update --suricata-conf ${cfg.configFile} --no-test \
-                --disable-conf ${pkgs.writeText "suricata-disable-conf" "${concatStringsSep "\n" cfg.disabledRules}"}
+                --disable-conf ${pkgs.writeText "suricata-disable-conf" "${concatStringsSep "\n" cfg.disabledRules}"} \
+                --enable-conf ${pkgs.writeText "suricata-enable-conf" "${concatStringsSep "\n" cfg.enabledRules}"}
             '';
           serviceConfig = {
             Type = "oneshot";
