diff --git a/nixos/modules/services/networking/syncthing.nix b/nixos/modules/services/networking/syncthing.nix
index adbb25ccb9b..b5c268ad827 100644
--- a/nixos/modules/services/networking/syncthing.nix
+++ b/nixos/modules/services/networking/syncthing.nix
@@ -22,6 +22,8 @@ let
     ignorePerms = folder.ignorePerms;
     ignoreDelete = folder.ignoreDelete;
     versioning = folder.versioning;
+    maxConflicts = 0;
+    ScanProgressIntervalS = -1;
   }) (filterAttrs (
     _: folder:
     folder.enable
