diff --git a/pkgs/by-name/ul/ulauncher/package.nix b/pkgs/by-name/ul/ulauncher/package.nix
index 3aed9d81f845..22c4e096f3de 100644
--- a/pkgs/by-name/ul/ulauncher/package.nix
+++ b/pkgs/by-name/ul/ulauncher/package.nix
@@ -9,7 +9,7 @@
   gobject-introspection,
   gtk3,
   wrapGAppsHook3,
-  webkitgtk_4_1,
+  webkitgtk_4_0,
   libnotify,
   keybinder3,
   libappindicator,
@@ -45,7 +45,7 @@ python3Packages.buildPythonApplication rec {
     libappindicator
     libnotify
     librsvg
-    webkitgtk_4_1
+    webkitgtk_4_0
     wmctrl
   ];
 
