diff --git a/nixos/modules/services/system/userborn.nix b/nixos/modules/services/system/userborn.nix
index bd3f2175b128..4d72229573ee 100644
--- a/nixos/modules/services/system/userborn.nix
+++ b/nixos/modules/services/system/userborn.nix
@@ -141,7 +141,7 @@ in
           ExecStart = "${lib.getExe cfg.package} ${userbornConfigJson} ${cfg.passwordFilesLocation}";
 
           ExecStartPre = lib.mkMerge [
-            (lib.mkIf (!config.system.etc.overlay.mutable) [
+            (lib.mkIf (cfg.passwordFilesLocation != "/etc") [
               "${pkgs.coreutils}/bin/mkdir -p ${cfg.passwordFilesLocation}"
             ])
 
