diff --git a/nixos/modules/services/web-apps/wordpress.nix b/nixos/modules/services/web-apps/wordpress.nix
index 6f44a7e84653..417d6db89819 100644
--- a/nixos/modules/services/web-apps/wordpress.nix
+++ b/nixos/modules/services/web-apps/wordpress.nix
@@ -30,6 +30,7 @@ let
         # symlink uploads directory
         ln -s ${cfg.uploadsDir} $out/share/wordpress/wp-content/uploads
         ln -s ${cfg.fontsDir} $out/share/wordpress/wp-content/fonts
+        ln -s ${cfg.cacheDir} $out/share/wordpress/wp-content/cache
 
         # https://github.com/NixOS/nixpkgs/pull/53399
         #
@@ -185,6 +186,15 @@ let
           '';
         };
 
+        cacheDir = mkOption {
+          type = types.path;
+          default = "/var/cache/wordpress/${name}";
+          description = ''
+            This directory is used to download fonts from a remote location, e.g.
+            to host google fonts locally.
+          '';
+        };
+
         plugins = mkOption {
           type =
             with types;
@@ -544,6 +554,8 @@ in
           "Z '${cfg.uploadsDir}' 0750 ${user} ${webserver.group} - -"
           "d '${cfg.fontsDir}' 0750 ${user} ${webserver.group} - -"
           "Z '${cfg.fontsDir}' 0750 ${user} ${webserver.group} - -"
+          "d '${cfg.cacheDir}' 0750 ${user} ${webserver.group} - -"
+          "Z '${cfg.cacheDir}' 0750 ${user} ${webserver.group} - -"
         ]) eachSite
       );
 
