diff --git a/kernel/Kbuild b/kernel/Kbuild
index a3efe41..5d41739 100644
--- a/kernel/Kbuild
+++ b/kernel/Kbuild
@@ -73,6 +73,7 @@ EXTRA_CFLAGS += -D__KERNEL__ -DMODULE -DNVRM -DNV_VERSION_STRING=\"510.108.03\"
 EXTRA_CFLAGS += $(call cc-option,-Werror=undef,)
 EXTRA_CFLAGS += -DNV_SPECTRE_V2=$(NV_SPECTRE_V2)
 EXTRA_CFLAGS += -DNV_KERNEL_INTERFACE_LAYER
+EXTRA_CFLAGS += -Wno-strict-prototypes
 
 #
 # Detect SGI UV systems and apply system-specific optimizations.
diff --git a/kernel/conftest.sh b/kernel/conftest.sh
index 1b0ad02..87fc745 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -102,7 +102,8 @@ test_header_presence() {
 build_cflags() {
     BASE_CFLAGS="-O2 -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
--nostdinc -isystem $ISYSTEM"
+-nostdinc -isystem $ISYSTEM \
+-Wno-implicit-function-declaration -Wno-strict-prototypes"
 
     if [ "$OUTPUT" != "$SOURCES" ]; then
         OUTPUT_CFLAGS="-I$OUTPUT/include2 -I$OUTPUT/include"
@@ -685,7 +686,7 @@ compile_test() {
             #include <linux/cpu.h>
             void conftest_register_cpu_notifier(void) {
                 register_cpu_notifier();
-            }" > conftest$$.c
+            }"
             compile_check_conftest "$CODE" "NV_REGISTER_CPU_NOTIFIER_PRESENT" "" "functions"
         ;;
 
@@ -701,7 +702,7 @@ compile_test() {
             #include <linux/cpu.h>
             void conftest_cpuhp_setup_state(void) {
                 cpuhp_setup_state();
-            }" > conftest$$.c
+            }"
             compile_check_conftest "$CODE" "NV_CPUHP_SETUP_STATE_PRESENT" "" "functions"
         ;;
 
@@ -2150,6 +2151,7 @@ compile_test() {
             }" > conftest$$.c
 
             $CC $CFLAGS -c conftest$$.c > /dev/null 2>&1
+            rm -f conftest$$.c
 
             if [ -f conftest$$.o ]; then
                 rm -f conftest$$.o
@@ -2159,7 +2161,7 @@ compile_test() {
             else
                 echo "#undef NV_DRM_UNIVERSAL_PLANE_INIT_HAS_FORMAT_MODIFIERS_ARG" | append_conftest "types"
 
-                echo "$CONFTEST_PREAMBLE
+                CODE="
                 #if defined(NV_DRM_DRMP_H_PRESENT)
                 #include <drm/drmP.h>
                 #endif
@@ -2179,19 +2181,10 @@ compile_test() {
                                 0,     /* unsigned int format_count */
                                 DRM_PLANE_TYPE_PRIMARY,
                                 NULL);  /* const char *name */
-                }" > conftest$$.c
-
-                $CC $CFLAGS -c conftest$$.c > /dev/null 2>&1
-
-                if [ -f conftest$$.o ]; then
-                    rm -f conftest$$.o
+                }"
 
-                    echo "#define NV_DRM_UNIVERSAL_PLANE_INIT_HAS_NAME_ARG" | append_conftest "types"
-                else
-                    echo "#undef NV_DRM_UNIVERSAL_PLANE_INIT_HAS_NAME_ARG" | append_conftest "types"
-                fi
+                compile_check_conftest "$CODE" "NV_DRM_UNIVERSAL_PLANE_INIT_HAS_NAME_ARG" "" "types"
             fi
-
         ;;
 
         vzalloc)
