diff --git a/pkgs/os-specific/linux/zfs/generic.nix b/pkgs/os-specific/linux/zfs/generic.nix
index 052353f31992..2bd8a77fcf7a 100644
--- a/pkgs/os-specific/linux/zfs/generic.nix
+++ b/pkgs/os-specific/linux/zfs/generic.nix
@@ -59,7 +59,6 @@ let
         optionals
         optional
         makeBinPath
-        versionAtLeast
         ;
 
       smartmon = smartmontools.override { inherit enableMail; };
@@ -72,7 +71,6 @@ let
         "user"
         "all"
       ];
-      isAtLeast22Series = versionAtLeast version "2.2.0";
 
       # XXX: You always want to build kernel modules with the same stdenv as the
       # kernel was built with. However, since zfs can also be built for userspace we
@@ -122,8 +120,7 @@ let
                                                         --replace-fail "/etc/default"            "$out/etc/default"
           substituteInPlace ./contrib/initramfs/Makefile.am \
             --replace-fail "/usr/share/initramfs-tools" "$out/usr/share/initramfs-tools"
-        ''
-        + optionalString isAtLeast22Series ''
+
           substituteInPlace ./udev/vdev_id \
             --replace-fail "PATH=/bin:/sbin:/usr/bin:/usr/sbin" \
              "PATH=${
@@ -139,23 +136,6 @@ let
           substituteInPlace ./config/zfs-build.m4 \
             --replace-fail "bashcompletiondir=/etc/bash_completion.d" \
               "bashcompletiondir=$out/share/bash-completion/completions"
-        ''
-        + optionalString (!isAtLeast22Series) ''
-          substituteInPlace ./etc/zfs/Makefile.am --replace-fail "\$(sysconfdir)/zfs" "$out/etc/zfs"
-
-          find ./contrib/initramfs -name Makefile.am -exec sed -i -e 's|/usr/share/initramfs-tools|'$out'/share/initramfs-tools|g' {} \;
-
-          substituteInPlace ./cmd/vdev_id/vdev_id \
-            --replace-fail "PATH=/bin:/sbin:/usr/bin:/usr/sbin" \
-            "PATH=${
-              makeBinPath [
-                coreutils
-                gawk
-                gnused
-                gnugrep
-                systemd
-              ]
-            }"
         '';
 
       nativeBuildInputs =
@@ -258,12 +238,6 @@ let
 
           # Remove tests because they add a runtime dependency on gcc
           rm -rf $out/share/zfs/zfs-tests
-
-          ${optionalString (lib.versionOlder version "2.2") ''
-            # Add Bash completions.
-            install -v -m444 -D -t $out/share/bash-completion/completions contrib/bash_completion.d/zfs
-            (cd $out/share/bash-completion/completions; ln -s zfs zpool)
-          ''}
         '';
 
       postFixup =
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index a9e14724b4ae..ff2e8d05c4ab 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -12982,7 +12982,7 @@ with pkgs;
     zfs_2_2
     zfs_2_3
     zfs_unstable;
-  zfs = zfs_2_2;
+  zfs = zfs_2_3;
 
   ### DATA
 
