diff --git a/pkgs/os-specific/linux/nvidia-x11/fabricmanager.nix b/pkgs/os-specific/linux/nvidia-x11/fabricmanager.nix
index f032ade0d837..3ec4a10c692a 100644
--- a/pkgs/os-specific/linux/nvidia-x11/fabricmanager.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/fabricmanager.nix
@@ -1,4 +1,4 @@
-nvidia_x11: sha256:
+sha256:
 
 {
   stdenv,
diff --git a/pkgs/os-specific/linux/nvidia-x11/generic.nix b/pkgs/os-specific/linux/nvidia-x11/generic.nix
index b74af83021d9..90b5eeeba1cb 100644
--- a/pkgs/os-specific/linux/nvidia-x11/generic.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/generic.nix
@@ -248,8 +248,9 @@ let
           openSha256;
         settings =
           if useSettings then
-            (if settings32Bit then pkgsi686Linux.callPackage else callPackage) (import ./settings.nix self settingsSha256)
+            (if settings32Bit then pkgsi686Linux.callPackage else callPackage) (import ./settings.nix settingsSha256)
               {
+                nvidia_x11 = self;
                 withGtk2 = preferGtk2;
                 withGtk3 = !preferGtk2;
                 fetchFromGitHub = fetchFromGithubOrNvidia;
@@ -257,14 +258,17 @@ let
         persistenced =
           if usePersistenced then
             lib.mapNullable
-              (hash: callPackage (import ./persistenced.nix self hash) {
+              (hash: callPackage (import ./persistenced.nix hash) {
+                nvidia_x11 = self;
                 fetchFromGitHub = fetchFromGithubOrNvidia;
               })
               persistencedSha256
           else { };
         fabricmanager =
           if useFabricmanager then
-            lib.mapNullable (hash: callPackage (import ./fabricmanager.nix self hash) { }) fabricmanagerSha256
+            lib.mapNullable (hash: callPackage (import ./fabricmanager.nix hash) {
+              nvidia_x11 = self;
+            }) fabricmanagerSha256
           else { };
         inherit persistencedVersion settingsVersion;
         compressFirmware = false;
diff --git a/pkgs/os-specific/linux/nvidia-x11/persistenced.nix b/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
index 805eaffab03a..8409252db8f7 100644
--- a/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
@@ -1,4 +1,4 @@
-nvidia_x11: sha256:
+sha256:
 
 {
   stdenv,
@@ -6,6 +6,7 @@ nvidia_x11: sha256:
   fetchFromGitHub,
   m4,
   libtirpc,
+  nvidia_x11,
 }:
 
 stdenv.mkDerivation rec {
diff --git a/pkgs/os-specific/linux/nvidia-x11/settings.nix b/pkgs/os-specific/linux/nvidia-x11/settings.nix
index df05a859b4c9..601109dc52ac 100644
--- a/pkgs/os-specific/linux/nvidia-x11/settings.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/settings.nix
@@ -1,4 +1,4 @@
-nvidia_x11: sha256:
+sha256:
 
 { stdenv
 , lib
@@ -17,6 +17,7 @@ nvidia_x11: sha256:
 , libXxf86vm
 , libvdpau
 , librsvg
+, nvidia_x11
 , wrapGAppsHook3
 , addDriverRunpath
 , withGtk2 ? false
