diff --git a/nixos/modules/services/security/fail2ban.nix b/nixos/modules/services/security/fail2ban.nix
index eebf2a2f7b5..cd2058ec94e 100644
--- a/nixos/modules/services/security/fail2ban.nix
+++ b/nixos/modules/services/security/fail2ban.nix
@@ -285,10 +285,6 @@ in
       }
     ];
 
-    warnings = mkIf (!config.networking.firewall.enable && !config.networking.nftables.enable) [
-      "fail2ban can not be used without a firewall"
-    ];
-
     environment.systemPackages = [ cfg.package ];
 
     environment.etc = {
